{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Overview\n",
    "------\n",
    "This notebook is intended to provide guidance for querying variants in the OpenCGA server through *pyopencga* from the studies that your user has access to; you can combine the filtration of variants with the exploration of their related clinical data (Samples, Individuals Genotypes etc.).<br>\n",
    "\n",
    "For guidance on how to loggin and get started with *opencga* you can refer to : [pyopencga_first_steps.ipynb](https://github.com/opencb/opencga/tree/develop/opencga-client/src/main/python/notebooks/user-training)<br>\n",
    "\n",
    "A good first step when start working with OpenCGA is to explore **Catalog**, which holds: information about our user, the projects and studies our user has permission to access and the clinical data from the studies. For guidance you can refer to : [pyopencga_catalog.ipynb](https://github.com/opencb/opencga/tree/develop/opencga-client/src/main/python/notebooks/user-training)<br>\n",
    "\n",
    "**[NOTE]** The server methods used by *pyopencga* client are defined in the following swagger URL:\n",
    "- https://ws.opencb.org/opencga-prod/webservices/\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Table of Contents:\n",
    "\n",
    "This Notebook is organised in the following sections:\n",
    "\n",
    "* [Querying Variants: Filter by Variant Annotation](#Querying-Variants:-Filter-by-Variant-Annotation)\n",
    "* [Querying Variants: Filter by Sample Data](#Querying-Variants:-Filter-by-Sample-Data)\n",
    "* [Aggregation Stats](#Aggregation-Stats)\n",
    "* [Use Cases](#Use-Cases)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Setup the Client and Login into *pyopencga* \n",
    "\n",
    "**Configuration and Credentials** \n",
    "\n",
    "Let's assume we already have *pyopencga* installed in our python setup (all the steps described on [001-pyopencga_first_steps.ipynb](https://github.com/opencb/opencga/tree/develop/opencga-client/src/main/python/notebooks/user-training)).\n",
    "\n",
    "You need to provide **at least** a host server URL in the standard configuration format for OpenCGA as a python dictionary or in a json file.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Logged succesfuly to https://ws.opencb.org/opencga-prod, your token is: eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJkZW1vdXNlciIsImF1ZCI6Ik9wZW5DR0EgdXNlcnMiLCJpYXQiOjE2MTg0OTM3MDQsImV4cCI6MTYxODQ5NzMwNH0.KtfSt8jD8liZIxlsSAJ7hXBopHMy3_tZ3mAbl9AeUS8 well done!\n"
     ]
    }
   ],
   "source": [
    "## Step 1. Import pyopencga dependecies\n",
    "from pyopencga.opencga_config import ClientConfiguration # import configuration module\n",
    "from pyopencga.opencga_client import OpencgaClient # import client module\n",
    "from pprint import pprint\n",
    "from IPython.display import JSON\n",
    "# import json\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import datetime\n",
    "\n",
    "## Step 2. OpenCGA host\n",
    "host = 'https://ws.opencb.org/opencga-prod'\n",
    "# host = 'http://localhost:1234/opencga'\n",
    "\n",
    "## Step 3. User credentials\n",
    "user = 'demouser'\n",
    "passwd = 'demouser' ## you can skip this, see below.\n",
    "####################################\n",
    "\n",
    "## Step 4. Create the ClientConfiguration dict\n",
    "config_dict = {'rest': {\n",
    "                       'host': host \n",
    "                    }\n",
    "               }\n",
    "\n",
    "## Step 5. Create the ClientConfiguration and OpenCGA client\n",
    "config = ClientConfiguration(config_dict)\n",
    "oc = OpencgaClient(config)\n",
    "\n",
    "## Step 6. Login to OpenCGA using the OpenCGA client \n",
    "# Pass the credentials to the client\n",
    "# (here we put only the user in order to be asked for the password interactively)\n",
    "# oc.login(user)\n",
    "\n",
    "# or you can pass the user and passwd\n",
    "oc.login(user, passwd)\n",
    "\n",
    "print('Logged succesfuly to {}, your token is: {} well done!'.format(host, oc.token))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Setup OpenCGA Variables\n",
    "\n",
    "Once we have defined a variable with the client configuration and credentials, we can access to all the methods defined for the client. These methods implement calls to query different data models in *OpenCGA*. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Define the OpenCGA variables to query data\n",
    "study = 'family:corpasome'\n",
    "\n",
    "## You can define list using string with commas\n",
    "genes='BRCA2'\n",
    "\n",
    "## or you can use python lists\n",
    "transcripts=['ENST00000530893']\n",
    "\n",
    "RESULT_SEPARATOR='\\n'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Querying Variants: Filter by Variant Annotation\n",
    "------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this section you will learn how to query variants filtering by **Variant Annotation**, this is one of the most useful and rich web service with many filter parameters.\n",
    "\n",
    "You can query variants using the following function:\n",
    "\n",
    "`\n",
    "  oc.variants.query()\n",
    "`\n",
    "\n",
    "OpenCGA implements probably the most advanced variant query engine. In this section you will learn the basixs about how to filter using the variant annotation. In the first sections you will learn how to filter using different parameters, then we will build more complex queries combining different parameters.\n",
    "\n",
    "**IMPORTANT NOTE**: Many of the calls in this notebook could potentially return thousands or millions of variants with all sample and annotation data, which can be fine for real life Python scripts. In order to avoid issues with Jupyter and unnecessary delays **all the examples here will execute small queries and limit the number of results returned**."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Filter by Region\n",
    "\n",
    "You can query by a region or a list of regions using the parameter `region`. The region must be of the format `chromosome:start-end`, single chromosome are also allowed, e.g. `14,15:21242091-23226874` would fetch all variants from chomosome 14 and a small region of chromosome 15."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by Region\n",
      "---------------------\n",
      "#Time: 937\n",
      "#Num matches: 360\n",
      "#Num results: 360\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "15:22830185:T:C\t,TUBGCP5\tupstream_gene_variant\n",
      "15:22833202:C:T\t,TUBGCP5\t2KB_upstream_variant\n",
      "15:22840279:G:T\t,TUBGCP5\tsynonymous_variant\n",
      "15:22849186:T:-\t,TUBGCP5\tnon_coding_transcript_exon_variant\n",
      "15:22853997:C:T\t,TUBGCP5\tintron_variant\n",
      "15:22862384:C:A\t,TUBGCP5\tintron_variant\n",
      "15:22864438:A:G\t,TUBGCP5\tintron_variant\n",
      "15:22866621:G:A\t,TUBGCP5\tintron_variant\n",
      "15:22867123:-:AAG\t,TUBGCP5\tnon_coding_transcript_exon_variant\n",
      "15:22869804:A:G\t,TUBGCP5\tintron_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter by a small region of chromsome 15\n",
    "variants = oc.variants.query(study=study, region='15:21242091-23226874', exclude='studies', limit=500)\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Region', limit=10) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "#JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since **pyopencga 2.0.1** implements a function to convert REST response to a Pandas object `.to_data_frame()`, this allows you store the results as a **pandas** dataframe object:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [],
   "source": [
    "## Convert REST response object 'variants' to Pandas datafrom\n",
    "df = variants.to_data_frame()\n",
    "\n",
    "formatted_df = df.drop(['names', 'studies'], 1)\n",
    "\n",
    "#print('The results can be stored and printed as a pandas DF:\\n\\n', formatted_df.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With the **Pandas dataframe** object, you can easily create plots using **maplotlib** or **seaborn** python libraries to show the results:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAcgAAAFxCAYAAAD3ZDhXAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuNCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8QVMy6AAAACXBIWXMAAAsTAAALEwEAmpwYAABPNElEQVR4nO3dd1gUV9sG8BuWJtJs2LuIRjRSYq/YUGn2iondJNY3xtdogsbYNRrFFnuNJihFsRMiliSWGAtiRcUGFopKW2B3vj94mY+BoZdd4f5dl5fs1OfMzO4zc+bMGR1BEAQQERGRhK6mAyAiItJGTJBEREQymCCJiIhkMEESERHJYIIkIiKSwQRJREQk44NNkLNnz4a1tTWsra3h6Oio6XBKlJeXl1h2a2trTYejVS5evCjZNhcvXtR0SEUuOjoa3333Hbp06YJmzZqJZfXy8iq2dfr4+Ei267Nnz4ptXUTaQi+nkePGjcO5c+cAAPr6+jh79iwqVqwoO+2CBQuwb98+8bO/vz+aNGlShKFqLw8PD1y6dAkA0KpVK+zZs0fDERWtixcvYtSoUVmG6+jooFy5cqhevTrs7e0xcuRIJuwSMG3aNPF4Kyi1Wo0zZ87g+PHjuHHjBl6/fg2lUglTU1NYWVnB0dERvXr1Qo0aNYooaipL3r17h19//RWhoaEIDQ1FeHg40h+5z+k30sfHB998802OyzY2Nsa///5b5DHLyTFBDhw4UEyQKSkpOHz4MD777LMs0yUnJ+Po0aPiZxsbm2JPjn369IGVlRUAwNTUtFjXpW3at28PY2NjTYcBQRCQkJCAsLAwhIWFwdfXF5s2bUKHDh00FlOdOnUwa9YsyefS5MWLF5LkaG9vjy5dukChUMDW1jZPy3j06BG++uor3Lp1K8u4mJgYXLp0CZcuXUJQUFCpO9mjkvH8+XOsXLlS02EUWo4J0tHRERUqVEBMTAwAwNfXVzZB/vHHH4iNjRU/DxgwoEiDzCguLg4mJibo1KkTOnXqVGzr0WZ2dnaws7PT2Prbt2+P9u3bIzExERcuXMDVq1cBpJ1ErVq1SqMJsnr16hg7dqzG1l/cnj9/Lvk8ZcoUtG3bNs/zh4eHY8SIEYiKihKH1axZE127doWlpSXi4uIQEhJSYlXT6d9nKn0MDAxgZWWFZs2a4c8//8x3tbyNjQ369OmTZbi+vn5RhZirHBOkgYEB3NzcsHPnTgDAnTt3cPv2bTRt2lQyna+vr/i3kZERXFxcAAAHDx7EhQsXcO/ePURHR+P9+/fQ19dHtWrV8Mknn+DTTz9Fw4YNJcvy8vLCunXrxM///vsvNm7ciGPHjiEyMhIuLi5YunQpZs+eLa63Zs2aCAoKEucJDAzEqVOncOfOHURFReHt27dQKBSoUqUKWrZsCQ8PD3z88ceS9Wa+tA8MDMT58+dx4MABPHr0COXKlUOHDh0wa9YsVK1aVTZWALh06ZKkmnHJkiXo378/1Go1Dhw4gKNHj+LBgwd4//49ypUrh4oVK6Jhw4Zo0aIFPvvsszxdGWZe7927d8W/M1f3rl69Gl5eXggKCkJMTAxq1KiBwYMHY+zYsdDR0cl1XXJsbW3FJDRx4kR0794dkZGRAIAHDx7IzhMREYHdu3fj/PnzePbsGVJSUlC1alW0bdsWY8aMQYMGDbLMExcXh3Xr1uH48eOIjo5GjRo1MGDAAIwZMwbNmjUTp5s8eTKmTJkCIGt18O7du9G6dWvJcoODg+Ht7Y3r168jJiYGBgYGqF27Njp37oxRo0ahcuXKkumLc5vmJxa56uuMJ6xyZc1s7ty5kuQ4ePBgeHp6ZvnRefnyJf74449slyMIAg4ePIh9+/YhLCxM9vsBAM+ePUO3bt3Ez0uWLIGpqSm2bduGu3fvIiEhQXL83rt3D3v27MGlS5cQGRkJQRBQtWpVfPLJJxg1alSWmqnM34V//vkHXl5eOHbsGN69e4cmTZpg2rRpaNeuHWJiYrB69WoEBgbi/fv3sLKywpQpU9C1a9cs5UtOTsahQ4dw7Ngx3Lt3T0zkVlZWcHJywuDBg2FgYCCZ5+HDh9i6dSuuXLmCyMhIqNVqWFhYoGrVqrCxsYGTk1OWk5moqCjs2bMHZ8+eRXh4OJRKJSwsLNC8eXMMGjQoS/uKzMf3rl278ODBA+zbtw/Pnj1DjRo14OHhgZEjR0KlUmHbtm3w9vZGREQEqlevjsGDB2PcuHGyx+n58+fx22+/4fr164iKioKBgQEaNGiA3r17Y/jw4ShXrpzssSCnUaNGuHr1qnhceXh45DtBWllZafxkN8cECaRVs6YnSCAtkcydO1f8/ObNG7EaFgB69OghVnn+8ssvWapxUlJS8PDhQzx8+BC+vr7YunVrjl/q8ePH48qVK3kuEAAcPnwYJ0+ezLLep0+f4unTpzh69CiWLVsGV1fXbJcxe/ZsyXqVSiUCAgIQGhoKf3//LF+O3MybNw+//fabZFhcXBzi4uLw5MkT/PHHH3B1dS3SqtOIiAj069cPr169EoeFh4djxYoVSEpKwuTJkwu9Dn19fVSqVElMkBUqVMgyTXBwMGbMmIH4+HjJ8GfPnsHb2xuHDx/GypUr0bNnT3FcQkICRo4cidu3b4vDHj9+jB9//LHA9x/UajXmzp0LHx8fyfCUlBTcuXMHd+7cwW+//YaNGzdmW11ZVNu0KGLJrxs3buDy5cvi5yZNmmD+/PlQKBRZpq1atSqGDh2a7bK+//57yfc+r9+PgwcP4p9//pEd9+uvv+KHH35ASkqKZPiTJ0/w5MkT+Pv7Y+7cuRg+fHi2cX366acICQkRP1+7dg3jxo3Djz/+iFWrVuHJkyfiuFu3buGLL77Ajh070KZNG3F4dHQ0xo0bl+W3KzY2FpcvX8bly5dx8OBBbN++XWyT8fDhQwwYMAAJCQmSeV6/fo3Xr18jJCQESqVSkiCvX7+OSZMmITo6Oss8QUFBCAoKgpubG5YuXQpdXfn2lCtWrJCU9/Hjx/jhhx8QExODu3fv4vTp05LtuHLlSiiVSslxKggCPD09s/w+paSk4ObNm7h58yb8/Pywc+dOVKpUSTaOzIriKu+PP/5A69atERcXB3Nzc3z00UcYOHAgnJycCr3svMo1QVpZWeHjjz/G9evXAQABAQGYNWuWuAGOHDmC1NRUcfqBAweKf1esWBFdu3ZFnTp1YGZmBn19fbx+/RqBgYGIiIhAcnIyfvjhBwQEBGS7/itXrqBFixZo164dkpOTYW5unmuhTE1N0a5dOzRs2BDm5uYwNDRETEwMzpw5g4cPH0KtVmPRokXo1asXDA0Ns11v27ZtYWtri8DAQNy7dw9A2hchMDAQffr0Ee8F7t+/H0+fPgUA1K5dG8OGDROX07x5c8THx+PQoUPisDZt2qB169ZITk7Gy5cvcfPmTdy/fz/XcuXX06dPYWhoiGHDhsHIyAj79+9HUlISAGDHjh2YOHFioQ7khIQEBAcH486dO+IwZ2dnyTTPnz/HtGnTkJiYCACoVasWevfuDUNDQ/z++++4ffs2lEolZs6ciaNHj6J27doAgLVr10qSY+PGjdGtWzeEh4fj2LFjBYp369atkoTUuHFjODo6IioqCn5+fkhJSUFMTAy++OILnDp1SvbedlFt04LEMmvWLDx58gQHDhwQ5xs6dKh4nzW3+61///235HO/fv1kk2NenDt3Ltfvh5x//vkHFhYW6NOnDypVqoSbN28CAK5evYr58+dDrVYDSDvRcnNzg66uLvz9/REVFYXU1FQsWLAAjRs3hoODg+zyQ0NDMXjwYBgbG2Pfvn1ISUmBSqXC9OnToVAoMGTIEBgYGGD//v1ITU2FWq3G1q1bJQly1qxZkuTYoUMHtGzZEjdv3kRwcDAA4Pbt25g5cya2b98OIO3CIT05mpmZoX///qhQoQLevHmDp0+fSk5MgLST488//1xMjnp6enB1dUW1atUQFBQkfqf8/f3RoEEDTJo0Sba8ISEh6NixI5o3bw5vb2+8fv0aAMSr6s6dO6NZs2Y4cOCAuK7Mx+n27dslybFTp05o2bIlYmJi4Ofnh/fv3+PevXv4+uuvxfKWhIy37aKionDu3DmcO3cObm5uWLZsWYFrwPIj1wQJpCW99AQZHR2N4OBgdO/eHYC0erV27dqSq8GtW7dCqVTi2rVrePLkCeLj41G9enW0bdtW/HG4f/++ePkvp2fPnlizZk22Z1ByFi1ahNTUVNy4cQOPHz/G+/fvUblyZXTu3BkPHz4EkLbxb968me0XrUePHvDy8oKOjg4+/fRTtGvXDiqVCkDamXifPn3Ee4FnzpwRE6TcPbB3796J8wLAypUrUaVKFck0kZGRsldfhbVq1SpxX1WvXh2LFy8GkPYFffjwYYFana5bty5L1bJCocCAAQMwbdo0yfC9e/eKybFKlSrw9/cX7zlNnDgRPXv2REREBJRKJXbv3o25c+ciNTUV3t7e4jJq164Nb29vGBkZAQDq1q2LjRs35itmtVot+XLXqVMHBw8eFE+QbG1tMWfOHABpx/ihQ4dk77cDhd+mBY1l7NixuHjxoiRB9unTJ9dq1XQRERGSz3LV2nmVl++HHBMTE/j6+mZpHbtjxw4xOSoUCuzbt0+8/TJkyBD07dsXqampEAQB27Zty/Z7O3XqVHz++ecA0mq3Mp58T58+HRMmTBDHHT9+HADEJA2k3a7IeGXs4uIiaWzy3//+F35+fgCACxcuIDQ0FB999BGUSqU4Te/evbO0xExJSZHUOvj6+kqquufPn49BgwYBAD7//HO4uLjg8ePHANIS2Pjx42VPZtq3b48tW7ZAR0cH1apVg6enpziuY8eO2Lx5M4C07973338PQHqcpp8gpBs2bBjmz58vWUb6Nrtw4YLsLbaipq+vj9atW6Nhw4YwMzPD/fv3cfr0afH48vf3h62treRCpLjkKUH26dMHixcvFn/ofHx80L17d9y6dUty/6B///6SrL5r1y6sXbsWcXFxOS4/MjIy2wQ5ceLEfCVHADh69CgWLVokOQCzW292hg0bJpbFwsJCPBsEgLdv3+YrHjMzM1hbW4vbytnZGc2bN0edOnXQoEED2NnZ4aOPPsrXMvPC0tJS/CEHgPr160vGv3v3rsjWZW9vjy+//DJL1VrGaurXr1/D3t4+22WkV709evRIcsz07dtXTI5A2glbfhPko0ePxMZm6cvMWHvg5uYGT09PsTbk6tWrsgmyKLZpUcWiSQX9fri7u8s+OpKx2rVly5aStgn16tWDnZ2deB84vVGYnPT2D0Ba24SMMtZu1K1bV/w7Y7yZq38zNzgcMGCAmCDTY/noo4/QqlUr7N69G0BaVfHNmzfRoEED1K1bF02aNEGbNm0k8WRcj0KhgJubm/jZwMAALi4u4nOtb9++RVhYGBo3bpylvM7OzuJ+yKm89erVk4xLP04fPXokqeLdv38/9u/fn2U9GeMuzgTZoUMH/PXXX1lqb4KDgzFx4kTxUZGDBw+WSILMU+YxMTGR1PuePXsW0dHRkqtHXV1d9O/fX/z8+++/Y/HixbkmRyDthnh28nuWGxoaipkzZ+aaHHNbb+aDLeMPf0Feoblq1SrxwIqNjcW5c+ewb98+/PDDD+jXrx/69++f5V5EYeVUBgDiGXt+tW/fHl999RVcXV3Fk5dLly5h+PDhkmoRIH8nE+lJI3OSsbS0lHzOfPWdF5njyrwMPT09yRV85unTFcU2LapY8qtatWqSz+m1KQVR0O9Hdt/njMdJ5kZSgHQb5XQSkrGBUOaq7ozHkZ7e/18bZIw38/GaOZbMn9P3TY8ePfD555+LJ3KhoaEICAjA+vXrMWXKFHTo0EHynHjG9ZiZmWU5jjKvJ7vvUV7Lm/nqM/04ze+xVdS/UZlZWlrK3tro3LmzJMmHhYUVaxzp8nQFCaSdtacnxJSUFPj4+EiqL9q3by/5Ama8T2RsbIw1a9agVatWMDIyQnBwsHjZnpv8Nlo5ceKEuPN1dHSwYsUKdO3aFSYmJnjw4AH69u2bp+Vk/AKlL6swGjVqBD8/Pzx8+FB8cDYsLAy///47kpKScOvWLaxYsQJLliwp1HoyyvyFKao6e1tbW3H/NWvWTIz5+fPn+OmnnyRVNBnvGdesWRMjRozIdrnly5cHkPaDkVHmk530+yz5YWFhkeMyUlNTJVd1madPVxTbtKhiya/MLSj9/PwwatSofNfQAAX/fmTXEtLc3Fzcz+lXohll3EaZj4+Mcrr/mznm7OLI6M2bN+Lz1nKxZdw36VW4169fR1hYGJ48eYKLFy/izp07UCqVWLRoETp16oTatWtL1vPu3TskJydLkmTm9WTX9iKnMuXl/nLmY6tXr15ZWvhn1LJly1yXWRJK4v4jkI+u5hwcHCQZfN26dZIvccbGOQAk42rVqoVOnTqJZ1cZOxUoahnXa2pqir59+4r3vIprvRkP0vRq6Mxu3boFQRDQoEEDODs748svv8SqVavE+w4AJK3RPhQeHh5o1KiR+Nnb21u8HwtAUqX65s0bdOnSBWPHjs3yz97eXvxiNmjQQPJs3IkTJyQtGw8ePJjvOOvXry+5Kjt69KjkvpG/v7+ksVlxPmeqqVhatGghuXd3+/ZtLFiwQLKudC9fvpTc6yxuGct47do1yRXC48ePJdWqxblvMi87Y+M6uc/p0z99+hRv376FsbEx2rZti5EjR2LOnDnYtWuXOK1KpUJoaGiW9ahUKvj7+4ufk5OTceTIEfGzubl5lsfhikrmY/Ht27f49NNPs3w/R4wYgYoVK+Z4i6QoLFy4UNLSOF1wcLB4TxaQf+SpOOT5ChJIq3//8ccfAUgTQYUKFbI8r1O/fn1cuHABQNqzTdOnT4eVlRUuXbqUpTVdUcp4T+jdu3cYP3487O3tcevWLQQGBhbLOjNeOd+6dQsLFy4U76mOGDECRkZGGDlyJMzMzPDJJ5/A0tIS5ubmiIyMlFRT56WFrrZRKBSYOHEivv76awBpVz+bNm3CokWLAAAjR44UW3oqlUoMGjQIvXr1Qp06dZCamorw8HBcuXIFERERWLJkCZo2bQqFQoFBgwZhx44dANKqU4YMGYIuXbrg8ePHBTrR0dXVxejRo7Fq1SoAaU3eBw4ciG7duuHNmzeS+0oVKlSQ3C4oapqMZdGiRRg2bJhYVbZ//36cPXsWjo6OsLS0xPv37xESEoJLly7Bzs4ux0c9itLo0aMRGBgIQRCgUqkwYsQIuLu7Q0dHR3LCoKOjgzFjxhRbHE2aNEGHDh1w/vx5AGmt9GNiYtCyZUuEhITgzJkz4rRt27YV2w6cOnUKq1atgr29PerXry9WCWds8AP8/3e8X79+2Lhxo7gf5s+fj6tXr4qtWDMmg88++6zArY1zo6uri7Fjx4oNkf7++2+4uLigS5cuqFChAmJjY3H37l1cuXIFSUlJ6NevX56W+/btW2zatEn8nDHpPXnyBMuWLRM/T5o0Sdwufn5+2Lt3LxwcHNCyZUuUK1cO9+/fx6lTpyRV4TnVRBWlfCXIfv36Yc2aNVnOOF1dXbPUoX/66afw8/MT70EeP35cbDXWv3//LM9/FZUBAwZg165dYgOc8+fPiwd7ca3XyclJXK5arZZ0z9WvXz/xyjkyMlJyZpiRQqHAuHHjijy2ktC3b1+sW7cO4eHhANKugCZNmoTatWujVq1aWLNmDf7zn/8gPj4e8fHxedoHU6dOxd9//y0+6nHr1i2x6X2XLl0kP1R5rW4ZP348wsLCxLP1e/fuiY8npLOwsMD69etzrMYrCpqKpV69eti7dy9mzpwpXs08f/5c413K2dvbw9PTU2yBHhMTI54gpVMoFPjmm2/wySefFGssK1aswJgxY8RjL+NvSLrGjRtn6UotNTUVFy9ezLYXIjs7OzF2U1NTbNiwAZMmTUJsbCxSU1NlvxfOzs6YOHFiURQrW+PGjUN4eLjYcjz9OfXCeP/+fbaPhERGRkrGjRgxQnJxIAiC+LypnAkTJkgaYxWnfCXIKlWqoFOnTpJeawD5ruXq1KmDffv24ccff8SVK1cgCAKsra0xceJElC9fvtgSpLm5OX755ResWLECFy5cgFKpRP369TFq1Ci0bt26WNbbuXNnLFy4EHv37sXDhw9lG//MmzcPV65cwa1bt/D69WvExsZCV1cXlpaWaNmyJUaNGoUWLVoUeWwlQaFQYMKECWIHEikpKZKryC5duuDYsWPYt28fLly4gMePHyMpKQnGxsaoXbs2mjdvjs6dO6Njx47iMo2NjbF37154eXlJetJxd3eHm5ubJEHm9cpbV1cXy5cvh5OTEw4ePIgbN24gNjYW+vr6qF27Njp16oRRo0ZlaRRUHDQZS8OGDXHo0CEEBQXh+PHjuHnzpthZuZmZGaysrNC1a9cSfSAbAIYPHw47OzuxJ52XL19CEARYWlqiVatW8PDwKJbW3plVrFgRv/32G7y9vXHixAmxJ53y5cujUaNGcHJywpAhQyQtjx0dHZGYmIhr164hPDwc0dHRSExMhKmpKerXr4/u3btj5MiRkitBW1tbHD16FHv27EFwcDDCw8ORnJyMChUqwMbGBgMHDpS0mC4uOjo6WLhwIfr27Qtvb29cu3YNr1+/hiAIsLCwQMOGDfHJJ5+gR48exR7Ljh07cObMGfz111+IjIxEVFSUeAyk12iUZDebOkJBmmQSlYCkpCTJ4x3pdu7cKWnM9Ntvv+XYsICIqCDydQVJVJKcnZ1hZ2cHW1tbVKtWDe/evcOlS5ck9+js7OyYHImoWPAKkrRW+/btZZv8p2vcuDG2bt0qeRaMiKioMEGS1tq5cyeCg4MRFhaG2NhY8e0ITZo0Qa9eveDm5pbvTuOJiPKKCZKIiEhG/rvQICIiKgPYSCefoqLioFZ/2BfdFSoYIyYmIfcJP0As24eJZcu7KlWy9lVKxYNXkGWQnl7x9MqhDVi2DxPLRtqICZKIiEgGEyQREZEMJkgiIiIZTJBEREQymCCJiIhkMEESERHJYIIkIiKSwQRJREQkgwmSiIhIBhMkERGRDCZIIiIiGUyQREREMvg2DyoVyumlAilJSHmbhHJCatpAfSMkpvIQJ6KC4a8HlQ4pSYj95zSMyxsiIV4JALCw7wHomGg4MCL6ULGKlYiISAYTJBERkQwmSCIiIhlMkERERDKYIImIiGQwQRIREclggiQiIpLBBElERCSDCZKIiEgGEyQREZEMJkgiIiIZTJBEREQymCCJiIhkMEESERHJKNUJct26dbC2tsa9e/cAANeuXYOrqyt69eqFMWPGICoqSsMREhGRtiq1CfLWrVu4du0aatasCQBQq9X4+uuv4enpiZMnT8LBwQErV67UcJRERKStSmWCTE5OxoIFCzB//nxxWEhICAwNDeHg4AAAGDp0KE6cOKGhCImISNuVygS5Zs0auLq6olatWuKwiIgI1KhRQ/xcsWJFqNVqxMbGaiBCIiLSdnqaDqCo/fvvvwgJCcHMmTOLZfmVKpkUy3JLWpUqppoOoUilvE2CcXlDABD/NzTUg4l56SpnadtvGbFspG1KXYK8fPkywsLC0K1bNwBAZGQkxo4dCw8PD7x48UKcLjo6Grq6urCwsMjX8qOi4qBWC0UZcomrUsUUr1+/13QYRaqckIqEeCWMyxsiIV4JADBQpiK2FJWzNO63dCxb/pZHJaPUVbFOmDAB58+fR1BQEIKCglCtWjVs27YN48aNQ1JSEq5cuQIAOHDgAJycnDQcLRERaatSdwWZHV1dXSxfvhzz5s2DUqlEzZo1sWLFCk2HRUREWqrUJ8igoCDxbzs7Oxw5ckSD0RAR0Yei1FWxEhERFQUmSCIiIhlMkERERDKYIImIiGQwQRIREclggiQiIpLBBElERCSDCZKIiEgGEyQREZEMJkgiIiIZTJBEREQymCCJiIhkMEESERHJYIIkIiKSwQRJREQkgwmSiIhIBhMkERGRDCZIIiIiGUyQREREMpggiYiIZDBBEhERyWCCJCIiksEESUREJIMJkoiISAYTJBERkQwmSCIiIhlMkERERDKYIImIiGQwQRIREclggiQiIpLBBElERCSDCZKIiEgGEyQREZEMJkgiIiIZTJBEREQymCCJiIhkMEESERHJYIIkIiKSwQRJREQkgwmSiIhIBhMkERGRDCZIIiIiGUyQREREMpggiYiIZDBBEhERyWCCJCIiksEESUREJIMJkoiISAYTJBERkQwmSCIiIhlMkERERDKYIImIiGQwQRIREclggiQiIpLBBElERCSDCZKIiEgGEyQREZEMPU0HUFy++OILPHv2DLq6ujA2NsZ3332Hpk2b4tGjR5g9ezZiY2NhYWGBZcuWoV69epoOl4iItEypTZDLli2DqakpACAwMBBz5syBr68v5s2bh+HDh8PNzQ3+/v7w9PTE7t27NRwtERFpm1JbxZqeHAEgLi4OOjo6iIqKQmhoKJydnQEAzs7OCA0NRXR0tKbCJCIiLVVqryABYO7cubhw4QIEQcDWrVsRERGBqlWrQqFQAAAUCgUsLS0RERGBihUrajhaIiLSJqU6QS5atAgA4Ofnh+XLl2PatGmFXmalSiaFXoY2qFLFNPeJPiApb5NgXN4QAMT/DQ31YGJeuspZ2vZbRiwbaZtSnSDTubu7w9PTE9WqVcPLly+hUqmgUCigUqnw6tUrVK9ePc/LioqKg1otFGO0xa9KFVO8fv1e02EUqXJCKhLilTAub4iEeCUAwECZithSVM7SuN/SsWz5Wx6VjFJ5DzI+Ph4RERHi56CgIJibm6NSpUpo2rQpAgICAAABAQFo2rQpq1eJiCiLUnkFmZiYiGnTpiExMRG6urowNzfHpk2boKOjg/nz52P27NnYsGEDzMzMsGzZMk2HS0REWqhUJsjKlSvjt99+kx3XsGFDeHt7l3BERET0oSmVVaxERESFxQRJREQkQysTZHR0NOLj4wEAKpUKhw4dgq+vL9RqtYYjIyKiskIrE+TEiRMRHh4OAFi9ejW2b9+OnTt3YunSpRqOjIiIygqtTJCPHz9G06ZNAQCHDx/Gli1bsGvXLhw7dkzDkRERUVmhla1YdXV1kZKSgkePHsHU1BQ1atSAWq0Wq12JiIiKm1YmyI4dO2LatGmIjY1Fnz59AAAPHjxA1apVNRwZERGVFVqZIBcvXgxfX1/o6enB3d0dABATE4MpU6ZoNjAiIioztPIe5J49ezBkyBAMGDBAfPNG69at8erVKw1HRkREZYVWJsj169fLDt+4cWMJR0JERGWVVlWx/vXXXwAAtVqNv//+G4Lw/2/NePbsGcqXL6+p0IiIqIzRqgQ5d+5cAIBSqcScOXPE4To6OqhSpQq+/fZbTYVGRERljFYlyKCgIADArFmzsHz5cg1HQ0REZZlWJch0GZNj5u7ldHW18rYpERGVMlqZIG/duoUFCxbg7t27UCrT3g4vCAJ0dHRw+/ZtDUdHRERlgVYmyNmzZ6Nr165YvHgxjIyMNB0OERGVQVqZIJ8/f44ZM2ZAR0dH06EQEVEZpZU39Hr06IHz589rOgwiIirDtPIKUqlUYvLkybC3t0flypUl49i6lYiISoJWJshGjRqhUaNGmg6DiIjKMK1MkJMnT9Z0CEREVMZpZYJM73JOTtu2bUswEiIiKqu0MkGmdzmXLiYmBikpKahatSp+//13DUVFRERliVYmyPQu59KpVCps3LiRnZUTEWkhR0dHLFy4EO3atdN0KEVKKx/zyEyhUGDSpEnYunWrpkMhIqIy4oNIkABw4cIFdhxARKRlvv76a7x48QKTJk2Cra0tWrRogT179kimcXFxwenTpwEA1tbW2L17N7p164bWrVtj2bJlkj63Dx48iN69e+OTTz7B2LFj8fz58xItT0ZaWcXauXNnSTJMTExEcnIy5s2bp8GoiIgosxUrVuCff/4Rq1iPHTuGHTt2wMPDAwBw584dvHr1Cp07dxbnOX36NA4dOoSEhASMHj0aDRo0wKBBgxAYGIiff/4ZmzZtQt26dbF582Z89dVXOHDggEbKppUJcsWKFZLP5cqVQ/369WFiYqKhiIiIKC+6deuGefPm4fHjx6hXrx78/f3Ru3dvGBgYiNOMHz8eFhYWsLCwwKhRoxAQEIBBgwbhwIEDmDBhAho2bAgAmDRpEn7++Wc8f/4cNWvWLPGyaGWCbNWqFYC0V129efMGlStX5muuiIg+AIaGhujduzcOHz6MyZMnIyAgAGvXrpVMU716dfHvmjVr4tWrVwCAFy9eYPHixVi2bJk4XhAEvHz5kgkyXVxcHBYsWIBjx44hNTUVenp66Nu3L7799luYmppqOjwiIspBv379MGvWLNjb26NcuXKwtbWVjI+IiICVlRWAtKRoaWkJIC1xTpo0Ca6uriUesxytvCxbuHAhEhMTceTIEdy4cQNHjhxBYmIiFi5cqOnQiIgok8qVK+Pp06fiZ1tbW+jq6mLp0qWyyW7btm14+/YtIiIisHv3bvTp0wcAMHToUGzevBn3798HALx//x7Hjx8vmULI0MoEee7cOSxfvhz169eHgYEB6tevjyVLluDcuXOaDo2IiDKZMGECNm7cCAcHB2zbtg0A4Obmhnv37sHNzS3L9N26dUP//v3h7u6OLl26YODAgQDS3uQ0btw4/Oc//4GdnR2cnZ1x9uzZEi1LRlpZxWpoaIjo6GhJnXNMTIzkJi8REWmH7t27o3v37pJhNWrUgJ2dHWrXrp1l+s6dO2PUqFGyy3J3d4e7u3txhJlvWpkgBw4ciDFjxuCzzz5DjRo18OLFC+zcuRODBw/WdGhERJSLxMRE/PLLLxg+fLimQykUrUyQn3/+OSwtLREQEIBXr17B0tIS48aNEy/DiYhIO507dw5TpkxB27Zt4ezsrOlwCkVHEARB00F8SKKi4qBWf9ibrEoVU7x+/V7TYRSpckIcYv85DePyhkiIVwIALOx7IFGn9Dw7Wxr3WzqWLX/Lo5KhlY10Fi5ciKtXr0qGXb16FYsWLdJQREREVNZoZYIMCAiAjY2NZJiNjQ0CAgI0FBEREZU1WpkgdXR0kLnmV6VSSTq0JSIiKk5amSAdHBzw008/iQlRrVbDy8sLDg4OGo6MiIjKCq1sxTp37lxMnDgRHTp0QI0aNRAREYEqVapg06ZNmg6NiOiDp1KpoVAU/fVRfpZ7/Phx/PzzzxAEAUqlEs2aNcOPP/4IR0dHGBsb4/Dhw2If3I6Ojti0aRPOnDmD27dvY/Xq1ZJlzZ49G7Vq1cLkyZOLtDxamSCrVasGX19f3LhxAxEREahevTpatGjBDsuJiIqAQqGLX07eKfLlDu/VJE/TvXr1Ct9//z18fX1RvXp1CIKA27dvi+MTEhLg7++Pfv36SeZzd3fHxo0b8e7dO5iZmQEA4uPjcerUKRw5cqToCvI/WptxdHV10bJlS/Tu3RstW7ZkciQiKiXevHkDPT09WFhYAEhrd/LRRx+J4ydPnox169YhOTlZMp+lpSVat24tabB5/PhxfPzxx8Xytg9mHSIiKlFNmjRBixYt0KVLF0ydOhU7d+5ETEyMON7GxgbNmjXD/v37s8w7YMAA+Pj4iJ99fHwwYMCAYomTCZKIiEqUrq4uNmzYgD179qB169YIDg6Gq6srYmNjxWmmT5+OLVu2ID4+XjJvly5d8OLFC9y/fx+PHz/GgwcP0KNHj2KJUyvvQRIRUenXuHFjNG7cGCNGjECfPn1w6dIlcVyDBg3QuXNn7NixQzKPvr4+XF1d4ePjAwMDA/Tt2xeGhobFEp9WXkFm15N7//79SzYQIiIqci9fvsS///4rfo6MjER0dDRq1aolmW7KlCn45ZdfslxFDhgwAIcPH4afn1+xVa8CWnoFGR4enmWYIAh49uyZBqIhIqKilJqaCi8vLzx//hxGRkZQq9WYPn26pKEOkPZEg5ubG7Zv3y4ZbmVlhRo1akCpVGbpda0oaVVn5bNmzQIAHDt2THzDdLrnz59DEAT88ssvmghNxM7KtRM7K/+wsWz5W15hacNzkB8CrbqCrFOnjuzfAGBnZwcnJ6eSDomIqNQpriRWmpIjoGUJMr0XhI8//hgdO3bUcDRERFSWaVWCTNexY0c8fPgQd+7cQUJCgmQcX5pMREQlQSsT5KZNm7B+/Xo0adIERkZG4nAdHR0mSCIiKhFamSB37doFb29vNGmSt379iIiIippW3lE1MjJCgwYNNB0GERGVYVqZIKdNm4aFCxfi1atXUKvVkn9EREQlQSurWGfPng0A8Pb2FocJggAdHR3JK1GIiCj/BLUKOroKjS03/f2O27dvx9GjR3Hy5EnUqFEDQNrvv42NDUaOHAkfHx8sXrwYtWrVglKphL6+Pnr27Ilx48aJ7VMcHR1hYGAg6W5u/fr1qFWrlriexo0bF6g8Wpkgf//990LNHxMTg1mzZuHJkycwMDBA3bp1sWDBAlSsWBHXrl2Dp6cnlEolatasiRUrVqBSpUpFFDkRkfbT0VUg+uyvRb7cip2G5HueKlWqwMvLC0uWLJEd365dO6xduxYAEBUVhblz52L69OnYtGmTOM3atWsLnARzopVVrDVr1sz2X17o6Ohg3LhxOHnyJI4cOYLatWtj5cqVUKvV+Prrr+Hp6YmTJ0/CwcEBK1euLObSEBFRdoYOHYoLFy7gwYMHuU5bqVIlLFu2DH/99Rfu379f7LFp5RUkkHYVefnyZcTExCBjb3jLly/PdV4LCwu0bt1a/NyyZUvs378fISEhMDQ0hIODA4C0HdOtW7dsz1yIiKh4GRsbY+LEiVi9ejXWr1+f6/Tm5uaoW7cu7t+/DysrKwDA1KlTxSpWhUIheV9kYWhlgly3bh0OHDiAPn364MSJExgyZAgCAgKy9M+aF2q1Gvv374ejoyMiIiLEem4AqFixItRqNWJjY8U3W+emUqXS0bdnUfTnqE1S3ibBuHzaFyT9f0NDPZiYF7ycqqR4qJXSjip0DY2hMCpf8EALqbTtt4xYtrJr8ODB2LFjB65fv56n6TN3IV5cVaxamSAPHTqE7du3o3HjxvDx8cGcOXPg7OyMDRs25HtZP/zwA4yNjTFy5EicPn260LGxs3LtVE5IRUK8UtJZuYEyFbGFKGd6B+gZWdj3QOJ7zbSmLo37LR3Llr/llTb6+vqYMmUKVq1aherVq+c47du3b/HkyZNiSYiZaeU9yHfv3omF19fXR0pKClq0aIHLly/naznLli1DeHg4fvrpJ+jq6qJ69ep48eKFOD46Ohq6urp5vnokIqLi4eLigujoaMlLkzOLjo7GnDlz0LZtWzRq1KjYY9LKK8g6deqI9ctWVlbYv38/zMzMYG5unudlrFq1CiEhIdi8eTMMDAwAADY2NkhKSsKVK1fg4OCAAwcO8A0hRERaQFdXF//5z38wadIkyfA///wT7u7uSEpKgoGBAXr06IHx48dLpsl4DxIAFi5ciObNmwMARo8eDYXi/x89OXLkSJ5ziVa9DzJdcHAwjI2N8cknn+D69euYOXMmEhISMG/ePPTs2TPX+e/fvw9nZ2fUq1dPfFamVq1aWL9+Pa5evYp58+ZJHvOoXLlynmNjFat2Ko73QWZbxaqhd0yWxv2WjmXL3/IKS9PPQX4otPIKsnPnzuLfH3/8cb7vHVpZWeHu3buy4+zs7HDkyJFCxUdE9CErriRWmpIjoEUJ8tmzZ6hVqxYA4OnTp9lOV7t27ZIKiYiIyjCtSZAuLi74999/AQA9evSAjo5Olqa87GqOSIauLpJSVJJBRvoKgH0XExWK1iTI9OQIAHfu3NFgJEQflqQUFU78+UgyzKldfRgpdDQUEVHpoHWPeahUKnTv3h3JycmaDoWIiMowrUuQCoUCCoUCSUlJmg6FiIjKMK1LkAAwatQozJgxA5cuXcKTJ0/w9OlT8R8RERWOSq3KfaJiXm5KSgrWrFmDXr16wcXFBe7u7li6dCkuXLgAa2trbN68WZz24sWL6N+/PwCgT58+OHv2rGRZ8fHxsLOzw7Nnz4qmIP+jNfcgM/rhhx8AABcuXJAMZyMdIqLCU+gq8FtIQJEvd7CNc56n/eabb6BUKnHo0CGYmJggNTUVhw4dQnJyMqpUqYJdu3Zh6NChMDMzk8zXv39/+Pr6olOnTuKwEydOoHnz5uKTEEVFKxMkG+kQEZVejx8/RmBgIIKDg2Fiktbxhp6eHoYMGYKLFy/C0tIStra22LJlC7766ivJvO7u7li/fj3evn0r9ojj4+ODIUPy/y7K3GhlFSsREZVeoaGhqFu3bo5dvn3++ec4ePAgXr16JRleuXJltGnTBgEBaVfA4eHhuHfvXp56WcsvrbyCTE1NxS+//CL7Psh9+/ZpMDIiIioJlStXxuDBg7Fhwwb07t1bMm7AgAHYuHEjRowYAR8fH/Tp00fsVrQoaeUV5JIlS/Drr7/CwcEBt27dQs+ePREVFYU2bdpoOjQiIiqkjz76COHh4Xj79m2O040bNw6BgYF48uSJZHiXLl0QGRmJO3fuwN/fHwMGDCiWOLUyQZ46dQpbtmzBp59+CoVCgU8//RTr16/HxYsXNR0aEREVUr169eDo6AhPT0/ExcUBSHsG3tvbGwkJ//+SclNTU4wePRobN26UzK+npwdXV1fMnTsXpqamaNGiRbHEqZUJMikpSXxpppGRERITE9GwYUOEhoZqODIiIioKS5cuRb169TBgwAA4OzvDxcUFDx8+FF9PmG7kyJFQqbI+PjJw4ECEhIQU29UjoKX3IBs2bIibN2+iRYsWsLGxgZeXF0xMTFC1alVNh0ZE9MFTqVX5eiQjP8tV5PGNHgYGBpgxYwZmzJiRZVz79u3Fvw0NDREcHJxlmoYNG2b71qaiolVXkOr/da48Z84c8QWXs2fPRmhoKP744w/x+UgiIiq4vCYxbVmupmjVFWSnTp3g6uoKNzc3WFtbA0irq965c6dmAyMiojJHq64g58+fj2fPnmHQoEHo168fdu3ahejoaE2HRUREZZBWXUF2794d3bt3x7t373Ds2DH4+/tjxYoV6NChA/r16wdHR0fo6+trOkwiIioDtOoKMp2ZmRmGDh2K/fv34/jx47CxscGSJUvQoUMHTYdGRERlhFYmyHTJycm4efMmbty4gTdv3qBx48aaDomIiMoIrapiTXflyhX4+/vjxIkTqFixIlxdXTFv3jzUrFlT06EREVEZoVUJ0svLC4cPH0ZsbCycnJywadMm2NvbazosIiIqg7QqQV6/fh3Tp09H9+7dYWhoqOlwiIioDNOqBLl161ZNh0BERARAyxvpEBERaQoTJBERkQwmSCIiIhladQ+SSJukKMpBVdchyzCoNRQQEZUoJkiibCSlqHE86KZkmNvAatBTlK43FhCRPFaxEhERyWCCJCIiksEESUREJIMJkoiISAYTJBERkQwmSCIiIhlMkERERDKYIImIiGQwQRIREclggiQiIpLBBElERCSDCZKIiEgGOysn0jRdXSSlqCSDjPQVgFrmtSEy00KnGGMjKsOYIIk0LClFhRN/PpIMc2pXH0aKrJlPbtqebesVZ3hEZRarWImIiGQwQRIREclggiQiIpLBBElERCSDjXRIO+SnJWcRr6fY1lUI+nq6SEqVxvM6JoEtVolKEBMkaYX8tOQs6vUU17oKQ5miwqm/HkuGlS9viPYtqmsmIKIyiFWsREREMpggiYiIZDBBEhERyWCCJCIiksEESUREJIOtWMuguMQUJKkEyTBte8xBW+nq6SMplduu1Mjr40XZPR5koIek5NTc56cPEhNkGZSYlFIij1SURsoUFU7+FS4Zxm334crr40XZPR7Us229LI/j8HgoPVjFSkREJKNUJshly5bB0dER1tbWuHfvnjj80aNHGDJkCHr16oUhQ4bg8ePHmguSiIi0WqlMkN26dcO+fftQs2ZNyfB58+Zh+PDhOHnyJIYPHw5PT08NRUhERNquVCZIBwcHVK8u7ZIrKioKoaGhcHZ2BgA4OzsjNDQU0dHRmgiRiIi0XJlppBMREYGqVatCoVAAABQKBSwtLREREYGKFSvmeTmVKpkUV4iy4hJTkJiUIhlWzkgfJuX0C7zM1zEJKF/eUDLMyFAPVSoYF3iZcnECeY+1sDGlvE2C8f/mT//f0FAPJuamua4nu3W9fBUDfQPpV0RHB0W+7eRi0tXVlY1TbrjcMLmY8rOP8nrcFXa/Z1SlimnuExWxvB532R03ed32migbFV6ZSZBFJSoqDmq1kPuERSRJJci2skuMSyr4QvUUiI9XStejTMXr1+8LvEi5OIG8x5qkEgoVUzkhFQnxShiXN0TC/5ZjoExFbKb55daT3brUagEpmZrwCwKKZdtlXqZarc4yrHx5Q9nhcsPkYsrPPsrrcVfY/Z6uShXTQm3DgsrrcZfdcZOXbV/UZWOyLTmlsopVTvXq1fHy5UuoVGnPMqlUKrx69SpLVSwRERFQhhJkpUqV0LRpUwQEBAAAAgIC0LRp03xVrxIRUdlRKqtYFy5ciFOnTuHNmzcYPXo0LCwscPToUcyfPx+zZ8/Ghg0bYGZmhmXLlmk6VCIi0lKlMkF+++23+Pbbb7MMb9iwIby9vTUQERERfWjKTBUrERFRfpTKK0jScnIdPxdD15UpinJISsnU4rgku8jMa0fYGqavp4uk1EwxFcd2+kC2R7Eoy2X/gDFBUomT6/i5Z9t6xbAeNU78Ke1YvDjWk/3689YRtqYpU1RZOtwunv3xYWyP4lCWy/4hYxUrERGRDCZIIiIiGUyQREREMpggiYiIZDBBEhERyWArVqJCkn1MAijZR0pKQH4eBymxR0eKQWFjzzz/65iED6bsJMUESVRIco9JACX7SElJyM/jICX16EhxKGzsmecvX94Q7VvwpQgfIlaxEhERyWCCJCIiksEESUREJIMJkoiISAYb6VCpkKIoB1VdByTo6UL1vxaEOjqKPM8v13JRNx/zk4awE3AqRkyQVCokpahxPOgm9A30kJKcCgDoM6BGnueXa7nYq23dogyRigE7AafixCpWIiIiGUyQREREMpggiYiIZDBBEhERyWCCJCIiksFWrMVNphk6UExN0YujybvcMg30kPS/lqKiYmg0mF0n4HLr5yMZZYDcd4mNVakYMUEWM7lm6EDxNEUvjibvcsvs2bZeiXREnVMn4Hwko+zJ7lgkKi6sYiUiIpLBBElERCSDCZKIiEgGEyQREZEMNtL5AGXXulOuRZ/ctHo6WefNzzKp4GS3c0m1AC4j+zI/rZ/LyjahgmGC/ADl1LozL9M6tW9QqGVSwclt55JqAVxW9mV+Wj+XlW1CBcMqViIiIhlMkERERDKYIImIiGQwQRIREclggiQiIpLBVqwaItcUvVg6MC8hOgD0kZxluIG+LpJSyubjBlQI7JictAATpIbIPn5RDB2YlxwBSS8eZhmqTLbCyb/CJcPYtJ5yw47JSRuwipWIiEgGEyQREZEMJkgiIiIZTJBEREQymCCJiIhksBVrKSL3qMWH2iaWCofHAlHhMUGWKnKPWlhpJBLSNB4LRIXFKlYiIiIZTJBEREQymCCJiIhkMEESERHJYCMdLSLXgbmuTtltjZhdB+hlpfykGWwBTOmYILWIXAfmvdrWLcOtEeU7QC875SfNYAtgSsMqViIiIhlMkERERDKYIImIiGQwQRIREclggiQiIpLBVqzFTE9HLfuogi4ENiXPQL5pvVDky5Tb7unTFrWSelygOLZdYdadPjwv0+rrqFBOSELK2ySUE1LF4ak65Yo20GLCR0JKNybI4iao5R9VEBqxKbmETNN6oVEhFym/zJJ7dKSkHhcohm1XmHUDkC+nzLTq+oi9dhrG5Q2REK8UBytsnYs0yuLDR0JKM1axEhERySiTCfLRo0cYMmQIevXqhSFDhuDx48eaDomIiLRMmUyQ8+bNw/Dhw3Hy5EkMHz4cnp6emg6JiIi0TJm7BxkVFYXQ0FDs2LEDAODs7IwffvgB0dHRqFixYq7z6+rm7xa8ArowNTXOOlyRdbhCoQsTY/0CTZevZerqyKwn67DshssPy76cBZ0/f8tMm1ZPXw+pKalFusyClKew88sNMzbSL9S2y/64yes+Lux+l49JKFceuoYGUKj1cpm/4HEWtkyF+c5lv9908v17QiVLRxCEkmnupiVCQkLw3//+F0ePHhWH9enTBytWrECzZs00GBkREWmTMlnFSkRElJsylyCrV6+Oly9fQqVSAQBUKhVevXqF6tWrazgyIiLSJmUuQVaqVAlNmzZFQEAAACAgIABNmzbN0/1HIiIqO8rcPUgACAsLw+zZs/Hu3TuYmZlh2bJlaNCggabDIiIiLVImEyQREVFuylwVKxERUV4wQRIREclggiQiIpLBBElERCSDCfID98UXX8DV1RXu7u4YPnw4bt++DSDnDtkLOk4Tsiufo6MjnJyc4ObmBjc3N5w7d06c59q1a3B1dUWvXr0wZswYREVF5WmcJqxbtw7W1ta4d+9ervF9SOVKl7l81tbWcHFxEffb3bt3xWmDgoLg5OSEHj16YPr06UhMTMzTuJKW3bFX2vYdARDog/bu3Tvx79OnTwvu7u6CIAiCh4eH4OfnJwiCIPj5+QkeHh7idAUdpwnZla9r167C3bt3s0yvUqmE7t27C5cvXxYEQRDWr18vzJ49O9dxmhASEiKMHTtWLEtBY9e2cqXLXD5BEITGjRsLcXFxWaaNi4sT2rVrJzx69EgQBEGYM2eO4OXlles4TZA79krbvqM0vIL8wJmamop/x8XFQUdHR+yQ3dk57aWzzs7OCA0NRXR0dIHHaYpc+XISEhICQ0NDODg4AACGDh2KEydO5DqupCUnJ2PBggWYP3++OKygsWtTudLJlS8nZ8+ehY2NDerVqwcgrQzHjx/PdZy2KE37jv5fmXubR2k0d+5cXLhwAYIgYOvWrYiIiEDVqlWhUCgAAAqFApaWloiIiIAgCAUap8mehjKXL93MmTMhCALs7e3xn//8B2ZmZoiIiECNGjXEaSpWrAi1Wo3Y2Ngcx1lYWJRkkbBmzRq4urqiVq1a4rCCxq5N5UonV750Hh4eUKlU6NSpE6ZMmQIDA4MsZahRowYiIiIAZN0uGcdpSuZjrzTtO/p/vIIsBRYtWoQzZ85gxowZWL58uabDKXJy5du3bx8OHz6MQ4cOQRAELFiwQMNR5t2///6LkJAQDB8+XNOhFIucynfmzBn4+Phg3759ePDgAdavX6+BCAvnQz72KH+YIEsRd3d3XLx4EdWqVcu2Q/acOmvX9o7c08sXExMjxmRgYIDhw4fj6tWrANI6o3/x4oU4T3R0NHR1dWFhYZHjuJJ0+fJlhIWFoVu3bnB0dERkZCTGjh2L8PDwAsWuLeVKl135zp8/L+43ExMTDBo0KNv99uLFC3HanMZpgtyxV9D9o237jqSYID9g8fHxkqqmoKAgmJub59ghe0HHaUJ25TM0NMT79+8BAIIg4NixY2jatCkAwMbGBklJSbhy5QoA4MCBA3Bycsp1XEmaMGECzp8/j6CgIAQFBaFatWrYtm0bxo0bV6DYtaVc6bIrX/PmzZGUlAQASE1NxcmTJ8X91rFjR9y8eVNsNX3gwAH07t0713ElLSEhQfbYK+j+0bZ9R1Lsi/UD9ubNG3zxxRdITEyErq4uzM3N8d///hfNmjXLsUP2go7TlvKZmZlhypQpUKlUUKvVaNiwIb799ltYWloCAK5evYp58+ZBqVSiZs2aWLFiBSpXrpzrOE1xdHTEpk2b0Lhx4wLHro3lSpdevvj4eHh6ekJHRwepqamwtbXFnDlzUL58eQBAYGAgVqxYAbVajaZNm2Lp0qUwNjbOdVxJevr0abbHXmncd2UdEyQREZEMVrESERHJYIIkIiKSwQRJREQkgwmSiIhIBnvSISLKRK1WY9q0abh37x4MDQ1RqVIlfP/996hTp06WadevX49jx45BV1cX+vr6mDFjBjp27AggrfN/T09PvHv3DsnJyejTpw+mTJmS47qTk5MxaNAg8XNSUhKePn2KP//8k89HljRNdABLH461a9cKX331labDoDJu5MiRwm+//VZsy+/atavks0qlEgIDAwWVSiUIgiDs2bNHGDVqlOy8Z8+eFRISEgRBEITbt28L9vb2QmJioiAIgvD5558Le/bsEQQhrdP1Ll26CNevX89XbDt27BAmTJiQr3moaLCKlXDkyBH0798ftra26NChA8aNGyc+uFwQe/fuRf/+/WFjY4PZs2dLxj179gzW1tawtbUV/+XU3diyZcvQs2dP2NrawsnJCX5+fpLx3333HXr16oUmTZrAx8dHMs7X1xf9+/eHnZ0dOnXqhOXLlyM1NTXLOh4/fozmzZtj5syZ4rAzZ85g2LBhcHBwQPv27TF37lzExcWJ45OTk/HNN9/Azs4O7du3x44dOyTL9Pb2Ro8ePWBra4uxY8fi5cuX4ri///4bHh4esLe3h6OjY/Yb8n+2bt0KZ2dn2NrawtHRUdIfLZDWt2mbNm1gZ2cHV1dXBAYG5rkcuW1flUqF1atXo0OHDrC1tYW7uzvevXsHIO1B+dWrV6Njx46wt7eHh4cH7t+/L87bt29fyX7+6KOPMGnSpCzl8/Pzg7W1Nby9vXPdFiVFV1cX3bp1g65u2k9ky5YtJT3eZNSxY0eUK1cOQNrrvARBQGxsLABAR0dH7FggKSkJOjo6Yscbr169wtSpUzFw4EC4uLhg06ZNssv38fHBgAEDirJ4lFeaztCkWdu3bxfatGkjnDx5UoiPjxeSk5OF33//XVi6dKkgCPm7gkxJSREEQRBOnjwpnD59WvD09BT++9//SqZ5+vSp0LhxY3Ha3KxZs0Z48OCBoFKphGvXrgkODg7CP//8I47fu3ev8Oeffwr9+vUTDh06JJl33759wuXLlwWlUilERkYK/fr1E37++ecs6xg9erQwbNgwSTkPHz4sBAcHCwkJCUJsbKwwduxY4bvvvhPHr1y5Uhg2bJgQGxsrPHjwQGjXrp0QHBwsCIIg/P3330KbNm2Ee/fuCUqlUvD09BRGjBghznv9+nXB19dXOHDgQJYrFzmbN28WQkJChJSUFCEsLEzo0qWLEBAQII6/ffu2uD2vXbsmtGzZUnj58mWeypHb9l21apXg4eEhPHv2TFCr1cLdu3eFpKQkQRAE4ejRo0L79u2FJ0+eCKmpqcLKlSvF15Flplarha5duwq+vr6S4bGxsUKvXr2Evn375niFWNJXkJnNnj1bWLx4ca7L8fHxkWyDZ8+eCc7OzkKHDh2EFi1aCHv37hXHffbZZ8KlS5cEQRAEpVIpDBs2TDh//rxkeTdu3BDatGkjJCcn56c4VER4D7IMe//+PdauXYvFixejZ8+e4nBHR0fJlU1KSgpmzZqF06dPo0aNGli6dCmaN28uTjt06FAcOXIEjx49wrVr18Rl3bx5U3LlVBBTp04V//74449hb2+Pa9euwc7ODgAwYsQIAIChoWGWeTN2ll21alW4uLjg4sWLkmmOHj0KU1NT2NraIjw8XBzu4uIi/l2uXDkMHjwYXl5e4jBfX18sXboU5ubmMDc3x6BBg+Dr64tOnTrhzJkzcHJygpWVFYC0lz536tQJT548QZ06ddCiRQu0aNECf/75Z562wfjx48W/GzRogG7duuHq1avo27cvAKBJkybi+PReaiIjI2FpaZlrOXLavm/fvsXu3bvh7++PmjVrAgAaN24sTv/s2TPY29ujdu3aAABXV1fs3LlTtgyXL19GTEyM5DgDgB9//BEeHh55esXTkydPMHDgQDx8+BCtW7fGkiVLYGFhgWfPnqFbt25YsGCBWLbRo0dj7NixOS5v0qRJYleGr169gpubG4C0t9hkrI3YsmULwsLCsGvXrhyXd+nSJaxZswbbt28Xh/36669wc3PDuHHj8OrVK3h4eMDGxgZWVla4dOmS5FVy8fHxCAsLQ/v27cVhhw4dgqurK/T19XPdPlT0WMVahv37779QKpXo0aNHjtMFBQWhb9++uHLlChwdHfHDDz9Ixh89ehSbN2/GlStXoKeXt3Ourl27olOnTvjmm28kPxKbN2/GxIkTZedJSkpCSEgIGjVqlKd1ZHb58mXJvHFxcVi7di2++eabfM379u1bvH79WpKYmjRpggcPHoifBZkOqu7du1eguDMSBAFXrlzJsg0mTpyI5s2bY9CgQWjVqhVsbGxyLUdmmbfvvXv3oFAocOLECbRv3x69evXCvn37xOn79u2Lp0+f4tGjR0hJSYGvr6/YOCUzX19f9OrVS9I93I0bNxASEoJhw4blqex+fn5YvHgxzp8/Dz09PSxcuFAy/uLFizh16hS2bduGLVu25HoCsmnTJvj7+8Pf3x+Wlpbi3xmT4549exAQEIDNmzeL1ahy/v33X3z99ddYv369pGvGPXv2oF+/fgAAS0tLtGnTBpcvX4ZarYaOjg4OHjworjcwMBCjRo0S51UqlTh69CirVzWICbIMi42NRYUKFXJNavb29ujcuTMUCgXc3Nxw584dyXgPDw9Ur14dRkZGua6zQoUKOHjwIP744w/4+PggPj4eX3/9tTh+woQJ+Pnnn2XnnTdvHqytrbP9Ec7JwYMHERISgjFjxojDfvrpJwwYMADVqlXLcd4LFy7Az89PvNpKSEgAIH2Zs6mpKeLj4wGk3ZM6fvw47ty5g6SkJKxfvx46OjpiR92F4eXlBbVaneVH8+eff8bVq1exefNmdOjQQbx3llM5Msu8fSMjI/H+/Xs8fvwYv//+O9asWQMvLy9cuHABAFClShXY2dnByckJH3/8MU6cOCF7spGYmIiTJ0+KiQJIu7c5f/58eHp6ysYqx83NDY0bN4axsTGmTZuGEydOiG+eAYAvv/wSxsbGsLa2Rv/+/cVO9wvqwIED+O2337Bjx44cW4/euHEDM2bMwNq1a9GsWTPJuFq1auHcuXMA0k7I/vnnH1hZWcHExAT29vbYvHmzOG1ERARev34tfj516hTq1q0ruWqnksUEWYZZWFggJiZGtuFKRhk7TjYyMoJSqZTMk59XD5UvXx7NmzeHnp4eKleujO+++w7nz5+XNByRs2zZMty/fx9r1qyBjo5OntcHpHV0vWrVKmzZskVsIHH79m389ddf+Oyzz3Kc99q1a/jqq6+wdu1a1K9fHwDEq6CMMcfFxYmdbrdr1w5Tp07F1KlT4ejoiJo1a6J8+fK5JmIg7aomvVGLp6enZNzevXvh5+eHzZs3w8DAIMu8+vr66Ny5M86fP4/ff/8913JkJLd90094vvzySxgZGaFJkybo27cvgoODAaQ93hASEoLg4GDcuHEDkydPxqefforExETJsk+dOgULCwu0atVKHPbLL7/A2toaLVu2zHWbpMt4nNWoUQMpKSmIiYmRHV+zZk28evUqz8vOLC4uDvPnz0d8fDxGjx4NNzc3yaMX48ePx82bNwEA33//PZKSkuDp6Qk3Nze4ubnh7t27AIAlS5bgwIEDcHV1xeDBg+Hk5ITOnTsDAFauXImwsDC4uLjAxcUFM2bMEBtAAWycow14D7IMs7W1hYGBAQIDAwv1ip38Jiy5eeWqJNOtXbsW586dw549e2BiYpKv5Z89exbffvstNm/eDGtra3H4xYsX8fz5c3Tt2hVA2lWhSqVCv3794OvrCwAIDQ3F559/jsWLF6Nt27bivObm5qhSpQru3Lkj3i+6c+eOpOpyxIgR4v3RR48eYePGjeI9yZxMmjRJtqXnwYMHsXnzZuzbty/XRKtSqfD06VPxc3blSJfd9k3fXhn3b8a/79y5g969e4vx9O/fH4sXL8aDBw/Ee9RAWtWom5ubZN6//voLly9fxtmzZwGkVVuHhobi9u3bWU4M0mV89VlERAT09fVRoUIFcXhERAQaNmwIIO2dkelvd8mLoKAgyWcTE5MsNSUZbdmyRfz70KFD2U5nY2ODAwcOyI6rUqUKVq1ale28mVtGU8njFWQZZmpqiqlTp2LBggUIDAxEYmIiUlJSEBwcjOXLlxd4uampqVAqlVCr1VCpVJIrzuvXr+Phw4dQq9WIiYnBwoUL0apVK0l1ZUY///wzAgICsGPHDlSoUCHL+OTkZCiVSgiCIFkvkPYj/PXXX8PLywstWrSQzDdkyBCcPn0afn5+8PPzw9ChQ9GlSxds27YNQNr9t3HjxuG7776TfRTD3d0dGzduxNu3bxEWFgZvb2+xClGpVOLevXsQBAEvXryAp6cnRo0aBXNzcwBpD6ErlUqkpKRAEAQolUokJydnuz0PHz6M1atXY8eOHWKDmHRhYWEIDg5GUlISUlJS4O/vjytXruCTTz7JUzly2r516tSBg4MDNm3ahOTkZISFheHo0aPiSUXz5s1x4sQJvHnzBmq1Gn5+fkhNTUXdunXFZURGRuLixYuS6lUAWLp0KY4dOyZufxsbG0yePBkzZszIcTs8ePAAiYmJWLNmDXr16gWFQiGO37BhAxITE3H//n34+PigT58+2S6LKC94BVnGjRkzBpUrV8aGDRswc+ZMlC9fHs2aNZO9ismrjRs3Yt26deLnw4cPY/LkyZgyZQqePn2KVatWITo6GiYmJmjXrp3kLHrTpk24cuWK+KzfqlWroK+vL2n9OHHiRDG+sWPH4tKlSwDSGkp899132L17N1q3bo0NGzbg/fv3mDBhgjivvb09tm7dinLlykkaXRgbG8PAwECsgt2xYweio6Mxd+5czJ07F0Batd7Ro0cBpLX+nDdvHrp27QojIyOMHz8enTp1ApCWIL/66is8ffoU5cuXR//+/TFt2jRxXZcvX5Y0xmjRogVatWqFPXv2yG7Pn376CbGxsRg4cKA4zMXFBQsWLAAArFu3DtOnT4dCoUDdunWxevVq8V5YbuXIbfuuWrUKc+bMQevWrVGxYkVMmzZNvAodP348oqKi4O7ujoSEBNStWxdr166FmZmZuCx/f3+0bNkySw80GacB0qqHTUxMsj1RAtLuQc6ePRsPHz5Eq1atMH/+fMn4Vq1aoUePHhAEAWPGjEGHDh2yXRZRXvB9kET0QUt/zOPWrVt5bkVNlBesYiUiIpLBBElERCSDVaxEREQyeAVJREQkgwmSiIhIBhMkERGRDCZIIiIiGUyQREREMpggiYiIZPwfLmxJqwsLVk4AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432.725x360 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## Create a plot with the results:\n",
    "sns.set_theme(style=\"darkgrid\")\n",
    "sns.color_palette('deep')\n",
    "sns.displot(data=df, hue='type', hue_order=['SNV', 'INDEL', 'CNV'], x=\"start\", bins=50) # multiple= \"dodge\", 'stack'\n",
    "plt.title('Variants in Region of Chromosome 15', fontsize=20, fontweight='bold' )\n",
    "plt.xlabel('Chr15:21242091-23226874 bp', fontsize=12)\n",
    "plt.ylabel('Variant counts', fontsize=12)\n",
    "sns.despine(left=False)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Filter by Variant Type\n",
    "\n",
    "OpenCGA supports most existing variant types as you can see here:\n",
    "\n",
    " ```\n",
    "      SNV,                   // SO:0001483\n",
    "      MNV,                   // SO:0002007\n",
    "      INDEL,                 // SO:1000032\n",
    "      SV,                    // SO:0001537\n",
    "      INSERTION,             // SO:0000667\n",
    "      DELETION,              // SO:0000159\n",
    "      TRANSLOCATION,         // SO:0000199\n",
    "      INVERSION,             // SO:1000036\n",
    "      COPY_NUMBER,           // SO:0001019\n",
    "      COPY_NUMBER_GAIN,      // SO:0001742\n",
    "      COPY_NUMBER_LOSS,      // SO:0001743\n",
    "      DUPLICATION,           // SO:1000035\n",
    "      TANDEM_DUPLICATION,    // SO:1000173\n",
    "      BREAKEND,\n",
    "```\n",
    "You can filter by variant type(s) using the parameter `type`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by SNV\n",
      "------------------\n",
      "#Time: 242\n",
      "#Num matches: 288538\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:546952:T:C\t,RP5-857K21.4\tintron_variant\n",
      "1:745347:T:C\t,RP11-206L10.9,RP11-206L10.10,RP11-206L10.8\tnon_coding_transcript_exon_variant\n",
      "1:752566:G:A\t,RP11-206L10.10,FAM87B\tintron_variant\n",
      "1:752721:A:G\t,RP11-206L10.10,FAM87B\tintron_variant\n",
      "1:753405:C:A\t,RP11-206L10.10,FAM87B\tnon_coding_transcript_exon_variant\n",
      "\n",
      "\n",
      "Filter by SNV.INDEL\n",
      "------------------------\n",
      "#Time: 267\n",
      "#Num matches: 300709\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:546952:T:C\t,RP5-857K21.4\tintron_variant\n",
      "1:745347:T:C\t,RP11-206L10.9,RP11-206L10.10,RP11-206L10.8\tnon_coding_transcript_exon_variant\n",
      "1:745371:A:-\t,RP11-206L10.9,RP11-206L10.10,RP11-206L10.8\tnon_coding_transcript_exon_variant\n",
      "1:752566:G:A\t,RP11-206L10.10,FAM87B\tintron_variant\n",
      "1:752721:A:G\t,RP11-206L10.10,FAM87B\tintron_variant\n",
      "\n",
      "\n",
      "Filter by DELETION\n",
      "-----------------------\n",
      "#Time: 232\n",
      "#Num matches: 1\n",
      "#Num results: 1\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "12:133197450:GCCTGGGACTGACCCGGGCTCTCGAGGGGCCTCTCGTGTGCCCTTGTGACCCCCTTCCCTG:-\t,P2RX2,POLE\tintron_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Filter by SNV\n",
    "variants = oc.variants.query(study=study, type='SNV', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by SNV') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "# Filter by ...\n",
    "variants = oc.variants.query(study=study, type='SNV,INDEL', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by SNV.INDEL')\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "# Filter by ...\n",
    "variants = oc.variants.query(study=study, type='DELETION', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by DELETION')\n",
    "print(RESULT_SEPARATOR)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Filter by Gene\n",
    "\n",
    "We can filter variants by gene using the parameters `xref` or `gene`:\n",
    "* **xref**: you can filter by different IDs including gene, transcrit, dbSNP, ...\n",
    "* **gene**: only accepts gene IDs\n",
    "\n",
    "Remember you can pass different IDs using comma as separator."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by Transcripts: ['ENST00000530893']\n",
      "-----------------------------------------------\n",
      "#Time: 293\n",
      "#Num matches: 10\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "13:32884653:G:T\t,ZAR1L,BRCA2\tintron_variant\n",
      "13:32889968:G:A\t,ZAR1L,BRCA2\tintron_variant\n",
      "13:32890572:G:A\t,ZAR1L,BRCA2\t5_prime_UTR_variant\n",
      "13:32900933:T:A\t,BRCA2\tintron_variant\n",
      "13:32903685:C:T\t,BRCA2\tintron_variant\n",
      "\n",
      "\n",
      "Filter by Genes: BRCA2\n",
      "---------------------------\n",
      "#Time: 298\n",
      "#Num matches: 26\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "13:32884653:G:T\t,ZAR1L,BRCA2\tintron_variant\n",
      "13:32889968:G:A\t,ZAR1L,BRCA2\tintron_variant\n",
      "13:32890572:G:A\t,ZAR1L,BRCA2\t5_prime_UTR_variant\n",
      "13:32900933:T:A\t,BRCA2\tintron_variant\n",
      "13:32903685:C:T\t,BRCA2\tintron_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter by xref\n",
    "variants = oc.variants.query(study=study, xref=transcripts, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Transcripts: {}'.format(transcripts)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by gene\n",
    "variants = oc.variants.query(study=study, gene=genes, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Genes: {}'.format(genes))\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Filter by Consequence Type\n",
    "\n",
    "OpenCGA provides a rich variant annotation that includes Ensembl consequence types (https://m.ensembl.org/info/genome/variation/prediction/predicted_data.html). Yu can filter by consequence type by using parameter `ct`. You can provide a list od conseuqnce type names separated by comma. Also, an alias called `lof` filter by a combination of loss-of-function terms."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by missense_variant:\n",
      "--------------------------------\n",
      "#Time: 279\n",
      "#Num matches: 15401\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:69511:A:G\tOR4F5,\tmissense_variant\n",
      "1:909238:G:C\t,PLEKHN1,C1orf170\tmissense_variant\n",
      "1:909309:T:C\t,PLEKHN1,C1orf170\tmissense_variant\n",
      "1:914876:T:C\t,PLEKHN1,C1orf170\tmissense_variant\n",
      "1:916549:A:G\t,C1orf170\tmissense_variant\n",
      "\n",
      "\n",
      "Filter by missense_variant,stop_gained:\n",
      "--------------------------------------------\n",
      "#Time: 265\n",
      "#Num matches: 15578\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:69511:A:G\tOR4F5,\tmissense_variant\n",
      "1:909238:G:C\t,PLEKHN1,C1orf170\tmissense_variant\n",
      "1:909309:T:C\t,PLEKHN1,C1orf170\tmissense_variant\n",
      "1:914876:T:C\t,PLEKHN1,C1orf170\tmissense_variant\n",
      "1:916549:A:G\t,C1orf170\tmissense_variant\n",
      "\n",
      "\n",
      "Filter by lof:\n",
      "-------------------\n",
      "#Time: 297\n",
      "#Num matches: 5075\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:897730:C:T\tNOC2L,,PLEKHN1,KLHL17\tsplice_region_variant\n",
      "1:977330:T:C\t,AGRN\tsplice_region_variant\n",
      "1:987200:C:T\t,RP11-54O7.14,AGRN\tsplice_region_variant\n",
      "1:1233941:T:C\t,ACAP3\tsplice_region_variant\n",
      "1:1558792:T:C\tMIB2\tstart_lost\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "application/json": [
       {
        "alternate": "T",
        "annotation": {
         "alternate": "T",
         "chromosome": "1",
         "consequenceTypes": [
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000187961",
           "ensemblTranscriptId": "ENST00000338591",
           "geneName": "KLHL17",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "retained_intron",
           "cdnaPosition": 827,
           "cdsPosition": 0,
           "codon": "",
           "ensemblGeneId": "ENSG00000187961",
           "ensemblTranscriptId": "ENST00000463212",
           "geneName": "KLHL17",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001792",
             "name": "non_coding_transcript_exon_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000187583",
           "ensemblTranscriptId": "ENST00000379410",
           "geneName": "PLEKHN1",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000188976",
           "ensemblTranscriptId": "ENST00000327044",
           "geneName": "NOC2L",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000188976",
           "ensemblTranscriptId": "ENST00000477976",
           "geneName": "NOC2L",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000187583",
           "ensemblTranscriptId": "ENST00000379407",
           "geneName": "PLEKHN1",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000187961",
           "ensemblTranscriptId": "ENST00000481067",
           "geneName": "KLHL17",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000188976",
           "ensemblTranscriptId": "ENST00000469563",
           "geneName": "NOC2L",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000187583",
           "ensemblTranscriptId": "ENST00000379409",
           "geneName": "PLEKHN1",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "basic"
           ]
          },
          {
           "biotype": "nonsense_mediated_decay",
           "ensemblGeneId": "ENSG00000187961",
           "ensemblTranscriptId": "ENST00000466300",
           "geneName": "KLHL17",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "cds_start_NF",
            "mRNA_start_NF"
           ]
          },
          {
           "biotype": "processed_transcript",
           "ensemblGeneId": "ENSG00000188976",
           "ensemblTranscriptId": "ENST00000487214",
           "geneName": "NOC2L",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ]
          },
          {
           "ensemblGeneId": "",
           "ensemblTranscriptId": "",
           "geneName": "",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001566",
             "name": "regulatory_region_variant"
            }
           ]
          }
         ],
         "conservation": [
          {
           "description": "",
           "score": -0.4480000138282776,
           "source": "phylop"
          },
          {
           "description": "",
           "score": 0.05299999937415123,
           "source": "phastCons"
          },
          {
           "description": "",
           "score": 3.6700000762939453,
           "source": "gerp"
          }
         ],
         "cytoband": [
          {
           "chromosome": "1",
           "end": 2300000,
           "name": "p36.33",
           "stain": "gneg",
           "start": 1
          }
         ],
         "displayConsequenceType": "splice_region_variant",
         "end": 897730,
         "functionalScore": [
          {
           "description": "",
           "score": 0.4499998092651367,
           "source": "cadd_raw"
          },
          {
           "description": "",
           "score": 7.079999923706055,
           "source": "cadd_scaled"
          }
         ],
         "hgvs": [
          "ENST00000338591(ENSG00000187961):c.712-5C>T",
          "ENST00000463212(ENSG00000187961):n.827C>T"
         ],
         "populationFrequencies": [
          {
           "altAlleleFreq": 0.009202454,
           "population": "SAS",
           "refAlleleFreq": 0.9907975,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.16256611,
           "population": "AFR",
           "refAlleleFreq": 0.8374339,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.17473525,
           "population": "AFR",
           "refAlleleFreq": 0.82526475,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.08173077,
           "population": "JPT",
           "refAlleleFreq": 0.9182692,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.12037037,
           "population": "YRI",
           "refAlleleFreq": 0.8796296,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.0301343,
           "population": "ALL",
           "refAlleleFreq": 0.9698657,
           "study": "DISCOVEHR"
          },
          {
           "altAlleleFreq": 0.005050505,
           "population": "CEU",
           "refAlleleFreq": 0.9949495,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.027221963,
           "population": "EA",
           "refAlleleFreq": 0.972778,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.044293586,
           "population": "ALL",
           "refAlleleFreq": 0.9557064,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.056774642,
           "population": "EAS",
           "refAlleleFreq": 0.9432254,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.057539683,
           "population": "EAS",
           "refAlleleFreq": 0.9424603,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.045558088,
           "population": "OTH",
           "refAlleleFreq": 0.9544419,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.024464427,
           "population": "ALL",
           "refAlleleFreq": 0.9755356,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.047046587,
           "population": "FEMALE",
           "refAlleleFreq": 0.9529534,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.029785337,
           "population": "NFE",
           "refAlleleFreq": 0.97021466,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.16197184,
           "population": "AA",
           "refAlleleFreq": 0.8380282,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.025461668,
           "population": "TWINSUK_NODUP",
           "refAlleleFreq": 0.9745383,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.056242276,
           "population": "EAS",
           "refAlleleFreq": 0.9437577,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.02335236,
           "population": "ALSPAC",
           "refAlleleFreq": 0.9766476,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.12295082,
           "population": "ASW",
           "refAlleleFreq": 0.8770492,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.1764706,
           "population": "MSL",
           "refAlleleFreq": 0.8235294,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.05536683,
           "population": "AMR",
           "refAlleleFreq": 0.9446332,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.060606062,
           "population": "FIN",
           "refAlleleFreq": 0.93939394,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.058755226,
           "population": "EAS",
           "refAlleleFreq": 0.9412448,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.03846154,
           "population": "PUR",
           "refAlleleFreq": 0.96153843,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.051312648,
           "population": "AMR",
           "refAlleleFreq": 0.9486874,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.0070140283,
           "population": "ALL",
           "refAlleleFreq": 0.99298596,
           "study": "GONL"
          },
          {
           "altAlleleFreq": 0.19026549,
           "population": "GWD",
           "refAlleleFreq": 0.8097345,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.04945055,
           "population": "GBR",
           "refAlleleFreq": 0.9505494,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.023364486,
           "population": "IBS",
           "refAlleleFreq": 0.9766355,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.05703356,
           "population": "AMR",
           "refAlleleFreq": 0.94296646,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.0058139535,
           "population": "BEB",
           "refAlleleFreq": 0.99418604,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.16145833,
           "population": "ACB",
           "refAlleleFreq": 0.8385417,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.1919192,
           "population": "ESN",
           "refAlleleFreq": 0.8080808,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.029601702,
           "population": "NFE",
           "refAlleleFreq": 0.9703983,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.12765957,
           "population": "CLM",
           "refAlleleFreq": 0.87234044,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.040899795,
           "population": "OTH",
           "refAlleleFreq": 0.9591002,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.07110493,
           "population": "ALL",
           "refAlleleFreq": 0.92889506,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.009803922,
           "population": "ITU",
           "refAlleleFreq": 0.99019605,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.04418775,
           "population": "OTH",
           "refAlleleFreq": 0.9558123,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.031588085,
           "population": "NFE",
           "refAlleleFreq": 0.9684119,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.042244904,
           "population": "ALL",
           "refAlleleFreq": 0.9577551,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.03809524,
           "population": "CHS",
           "refAlleleFreq": 0.96190476,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.041198503,
           "population": "ALL",
           "refAlleleFreq": 0.9588015,
           "study": "MGP"
          },
          {
           "altAlleleFreq": 0.014732965,
           "population": "SAS",
           "refAlleleFreq": 0.98526704,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.0882353,
           "population": "PEL",
           "refAlleleFreq": 0.9117647,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.058252428,
           "population": "CHB",
           "refAlleleFreq": 0.94174755,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.16483408,
           "population": "AFR",
           "refAlleleFreq": 0.8351659,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.07747604,
           "population": "ALL",
           "refAlleleFreq": 0.922524,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.03125,
           "population": "MXL",
           "refAlleleFreq": 0.96875,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.034617458,
           "population": "FIN",
           "refAlleleFreq": 0.9653825,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.07348703,
           "population": "AMR",
           "refAlleleFreq": 0.92651296,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.004854369,
           "population": "GIH",
           "refAlleleFreq": 0.9951456,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.056074765,
           "population": "TSI",
           "refAlleleFreq": 0.94392526,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.072857365,
           "population": "ALL",
           "refAlleleFreq": 0.9271426,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.02562028,
           "population": "TWINSUK",
           "refAlleleFreq": 0.9743797,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.03696275,
           "population": "FIN",
           "refAlleleFreq": 0.96303725,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.024509804,
           "population": "STU",
           "refAlleleFreq": 0.9754902,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.06989247,
           "population": "CDX",
           "refAlleleFreq": 0.93010753,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.028924774,
           "population": "ASJ",
           "refAlleleFreq": 0.97107524,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.03830094,
           "population": "MALE",
           "refAlleleFreq": 0.96169907,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.06638704,
           "population": "FEMALE",
           "refAlleleFreq": 0.93361294,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.04040404,
           "population": "KHV",
           "refAlleleFreq": 0.959596,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.07492677,
           "population": "MALE",
           "refAlleleFreq": 0.9250732,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.023178808,
           "population": "ASJ",
           "refAlleleFreq": 0.9768212,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.24242425,
           "population": "LWK",
           "refAlleleFreq": 0.75757575,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.16425565,
           "population": "AFR",
           "refAlleleFreq": 0.8357444,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.032801963,
           "population": "FIN",
           "refAlleleFreq": 0.967198,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.038767397,
           "population": "EUR",
           "refAlleleFreq": 0.9612326,
           "study": "1kG_phase3"
          }
         ],
         "reference": "C",
         "repeat": [],
         "start": 897730,
         "traitAssociation": [],
         "variantTraitAssociation": {},
         "xrefs": [
          {
           "id": "ENST00000463212",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000379410",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000327044",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000379407",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000481067",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000469563",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000379409",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000466300",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000463212(ENSG00000187961):n.827C>T",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000487214",
           "source": "ensemblTranscript"
          },
          {
           "id": "rs7549631",
           "source": "dbSNP"
          },
          {
           "id": "ENST00000338591",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENSG00000187583",
           "source": "ensemblGene"
          },
          {
           "id": "ENST00000477976",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENSG00000187961",
           "source": "ensemblGene"
          },
          {
           "id": "ENSG00000188976",
           "source": "ensemblGene"
          },
          {
           "id": "ENST00000338591(ENSG00000187961):c.712-5C>T",
           "source": "ensemblTranscript"
          }
         ]
        },
        "chromosome": "1",
        "end": 897730,
        "id": "1:897730:C:T",
        "length": 1,
        "names": [],
        "reference": "C",
        "start": 897730,
        "strand": "+",
        "studies": [],
        "type": "SNP"
       },
       {
        "alternate": "C",
        "annotation": {
         "alternate": "C",
         "chromosome": "1",
         "consequenceTypes": [
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000479707",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000379370",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "processed_transcript",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000477585",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0002083",
             "name": "2KB_downstream_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000469403",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0002083",
             "name": "2KB_downstream_variant"
            }
           ]
          },
          {
           "ensemblGeneId": "",
           "ensemblTranscriptId": "",
           "geneName": "",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001566",
             "name": "regulatory_region_variant"
            }
           ]
          }
         ],
         "conservation": [
          {
           "description": "",
           "score": -0.8539999723434448,
           "source": "phylop"
          },
          {
           "description": "",
           "score": 0.003000000026077032,
           "source": "phastCons"
          },
          {
           "description": "",
           "score": -2.7699999809265137,
           "source": "gerp"
          }
         ],
         "cytoband": [
          {
           "chromosome": "1",
           "end": 2300000,
           "name": "p36.33",
           "stain": "gneg",
           "start": 1
          }
         ],
         "displayConsequenceType": "splice_region_variant",
         "end": 977330,
         "functionalScore": [
          {
           "description": "",
           "score": -0.6400003433227539,
           "source": "cadd_raw"
          },
          {
           "description": "",
           "score": 0.10000000149011612,
           "source": "cadd_scaled"
          }
         ],
         "geneTraitAssociation": [
          {
           "hpo": "HP:0003388",
           "id": "OMIM:614198",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616321",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:254300",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0003402",
           "id": "OMIM:608930",
           "name": "Decreased miniature endplate potentials"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:254210",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:601462",
           "name": "High palate"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:608930",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:254300",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0003443",
           "id": "OMIM:608931",
           "name": "Decreased size of nerve terminals"
          },
          {
           "hpo": "HP:0000276",
           "id": "OMIM:608931",
           "name": "Long face"
          },
          {
           "hpo": "HP:0002872",
           "id": "OMIM:254210",
           "name": "Apneic episodes precipitated by illness, fatigue, stress"
          },
          {
           "hpo": "HP:0001260",
           "id": "OMIM:601462",
           "name": "Dysarthria"
          },
          {
           "hpo": "HP:0001263",
           "id": "OMIM:616330",
           "name": "Global developmental delay"
          },
          {
           "hpo": "HP:0002033",
           "id": "OMIM:254210",
           "name": "Poor suck"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:608931",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0001270",
           "id": "OMIM:608931",
           "name": "Motor delay"
          },
          {
           "hpo": "HP:0008180",
           "id": "OMIM:254300",
           "name": "Mildly elevated creatine phosphokinase"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:608931",
           "name": "High palate"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:608930",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0001260",
           "id": "OMIM:608930",
           "name": "Dysarthria"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:605809",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:608931",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0002804",
           "id": "OMIM:254210",
           "name": "Arthrogryposis multiplex congenita"
          },
          {
           "hpo": "HP:0001283",
           "id": "OMIM:605809",
           "name": "Bulbar palsy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:605809",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0000276",
           "id": "OMIM:616313",
           "name": "Long face"
          },
          {
           "hpo": "HP:0002515",
           "id": "OMIM:254300",
           "name": "Waddling gait"
          },
          {
           "hpo": "HP:0001612",
           "id": "OMIM:254210",
           "name": "Weak cry"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:616323",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0001270",
           "id": "OMIM:614198",
           "name": "Motor delay"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:616324",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616323",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000544",
           "id": "OMIM:614198",
           "name": "External ophthalmoplegia"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:605809",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:601462",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0003403",
           "id": "OMIM:608931",
           "name": "EMG: decremental response of compound muscle action potential to repetitive nerve stimulation"
          },
          {
           "hpo": "HP:0000486",
           "id": "OMIM:608931",
           "name": "Strabismus"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:605809",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:608931",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0003828",
           "id": "OMIM:608931",
           "name": "Variable expressivity"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:616322",
           "name": "High palate"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:608931",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616326",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:616326",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0001283",
           "id": "OMIM:254210",
           "name": "Bulbar palsy"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616323",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:601462",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0003593",
           "id": "OMIM:608930",
           "name": "Infantile onset"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:608930",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0001288",
           "id": "OMIM:616040",
           "name": "Gait disturbance"
          },
          {
           "hpo": "HP:0001260",
           "id": "OMIM:608931",
           "name": "Dysarthria"
          },
          {
           "hpo": "HP:0001270",
           "id": "OMIM:608930",
           "name": "Motor delay"
          },
          {
           "hpo": "HP:0003324",
           "id": "OMIM:601462",
           "name": "Generalized muscle weakness"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:605809",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616325",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616323",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:614198",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002098",
           "id": "OMIM:605809",
           "name": "Respiratory distress"
          },
          {
           "hpo": "HP:0001612",
           "id": "OMIM:605809",
           "name": "Weak cry"
          },
          {
           "hpo": "HP:0002715",
           "id": "OMIM:608931",
           "name": "Abnormality of the immune system"
          },
          {
           "hpo": "HP:0003693",
           "id": "OMIM:254300",
           "name": "Distal amyotrophy"
          },
          {
           "hpo": "HP:0000467",
           "id": "OMIM:616325",
           "name": "Neck muscle weakness"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616304",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:614198",
           "name": "High palate"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:608931",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:608931",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0007126",
           "id": "OMIM:254300",
           "name": "Proximal amyotrophy"
          },
          {
           "hpo": "HP:0000467",
           "id": "OMIM:608930",
           "name": "Neck muscle weakness"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:254210",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:605809",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616314",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:254210",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616326",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0000303",
           "id": "OMIM:608931",
           "name": "Mandibular prognathia"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616326",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:616326",
           "name": "High palate"
          },
          {
           "hpo": "HP:0003202",
           "id": "OMIM:616040",
           "name": "Skeletal muscle atrophy"
          },
          {
           "hpo": "HP:0001324",
           "id": "OMIM:616314",
           "name": "Muscle weakness"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:616323",
           "name": "High palate"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:608930",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0001761",
           "id": "OMIM:616040",
           "name": "Pes cavus"
          },
          {
           "hpo": "HP:0003394",
           "id": "OMIM:608931",
           "name": "Muscle cramps"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616325",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:254300",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616322",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616325",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616330",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616304",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002882",
           "id": "OMIM:254210",
           "name": "Sudden episodic apnea"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616313",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:608931",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0002650",
           "id": "OMIM:608930",
           "name": "Scoliosis"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:608931",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0003202",
           "id": "OMIM:616313",
           "name": "Skeletal muscle atrophy"
          },
          {
           "hpo": "HP:0001324",
           "id": "OMIM:616330",
           "name": "Muscle weakness"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:616326",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0000347",
           "id": "OMIM:608930",
           "name": "Micrognathia"
          },
          {
           "hpo": "HP:0003593",
           "id": "OMIM:608931",
           "name": "Infantile onset"
          },
          {
           "hpo": "HP:0003621",
           "id": "OMIM:254300",
           "name": "Juvenile onset"
          },
          {
           "hpo": "HP:0001371",
           "id": "OMIM:616313",
           "name": "Flexion contracture"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:614198",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:615120",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001251",
           "id": "OMIM:616330",
           "name": "Ataxia"
          },
          {
           "hpo": "HP:0002804",
           "id": "OMIM:608930",
           "name": "Arthrogryposis multiplex congenita"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616324",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:601462",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0001283",
           "id": "OMIM:254300",
           "name": "Bulbar palsy"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:254210",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:608930",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0002098",
           "id": "OMIM:254210",
           "name": "Respiratory distress"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616313",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616322",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0000486",
           "id": "OMIM:605809",
           "name": "Strabismus"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616325",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:601462",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0003403",
           "id": "OMIM:254210",
           "name": "EMG: decremental response of compound muscle action potential to repetitive nerve stimulation"
          },
          {
           "hpo": "HP:0003674",
           "id": "OMIM:601462",
           "name": "Onset"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616324",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0003397",
           "id": "OMIM:605809",
           "name": "Generalized hypotonia due to defect at the neuromuscular junction"
          },
          {
           "hpo": "HP:0002355",
           "id": "OMIM:616304",
           "name": "Difficulty walking"
          },
          {
           "hpo": "HP:0003397",
           "id": "OMIM:254210",
           "name": "Generalized hypotonia due to defect at the neuromuscular junction"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616322",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616322",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616325",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:601462",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0001371",
           "id": "OMIM:616330",
           "name": "Flexion contracture"
          },
          {
           "hpo": "HP:0002804",
           "id": "OMIM:608931",
           "name": "Arthrogryposis multiplex congenita"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:254300",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0003199",
           "id": "OMIM:608931",
           "name": "Decreased muscle mass"
          },
          {
           "hpo": "HP:0000365",
           "id": "OMIM:616040",
           "name": "Hearing impairment"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616322",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:608930",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:254210",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0000467",
           "id": "OMIM:616324",
           "name": "Neck muscle weakness"
          },
          {
           "hpo": "HP:0003577",
           "id": "OMIM:254210",
           "name": "Congenital onset"
          },
          {
           "hpo": "HP:0003828",
           "id": "OMIM:608930",
           "name": "Variable expressivity"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:254300",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:614198",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:616325",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:616322",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616330",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616323",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:608931",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616324",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:616325",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:615120",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0001260",
           "id": "OMIM:616330",
           "name": "Dysarthria"
          },
          {
           "hpo": "HP:0002033",
           "id": "OMIM:608930",
           "name": "Poor suck"
          },
          {
           "hpo": "HP:0003690",
           "id": "OMIM:605809",
           "name": "Limb muscle weakness"
          },
          {
           "hpo": "HP:0001612",
           "id": "OMIM:608930",
           "name": "Weak cry"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616322",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0003436",
           "id": "OMIM:601462",
           "name": "Prolonged miniature endplate currents"
          },
          {
           "hpo": "HP:0001612",
           "id": "OMIM:608931",
           "name": "Weak cry"
          },
          {
           "hpo": "HP:0000006",
           "id": "OMIM:601462",
           "name": "Autosomal dominant inheritance"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616324",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616324",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0003701",
           "id": "OMIM:616040",
           "name": "Proximal muscle weakness"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616321",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:254300",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0004889",
           "id": "OMIM:601462",
           "name": "Intermittent episodes of respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0003577",
           "id": "OMIM:605809",
           "name": "Congenital onset"
          },
          {
           "hpo": "HP:0001290",
           "id": "OMIM:605809",
           "name": "Generalized hypotonia"
          },
          {
           "hpo": "HP:0003680",
           "id": "OMIM:608931",
           "name": "Nonprogressive"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616313",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002033",
           "id": "OMIM:605809",
           "name": "Poor suck"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:254300",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0003324",
           "id": "OMIM:616321",
           "name": "Generalized muscle weakness"
          },
          {
           "hpo": "HP:0000689",
           "id": "OMIM:608931",
           "name": "Dental malocclusion"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:615120",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0003202",
           "id": "OMIM:608931",
           "name": "Skeletal muscle atrophy"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616321",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002104",
           "id": "OMIM:614198",
           "name": "Apnea"
          },
          {
           "hpo": "HP:0001288",
           "id": "OMIM:614198",
           "name": "Gait disturbance"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:608931",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:608931",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0002804",
           "id": "OMIM:616326",
           "name": "Arthrogryposis multiplex congenita"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616323",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616326",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0002872",
           "id": "OMIM:605809",
           "name": "Apneic episodes precipitated by illness, fatigue, stress"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:608930",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:254210",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616314",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0002355",
           "id": "OMIM:616330",
           "name": "Difficulty walking"
          },
          {
           "hpo": "HP:0003307",
           "id": "OMIM:614198",
           "name": "Hyperlordosis"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:616330",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616314",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:616313",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:608930",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0002882",
           "id": "OMIM:605809",
           "name": "Sudden episodic apnea"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:601462",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002540",
           "id": "OMIM:616321",
           "name": "Inability to walk"
          },
          {
           "hpo": "HP:0000774",
           "id": "OMIM:615120",
           "name": "Narrow chest"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616313",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0000276",
           "id": "OMIM:616326",
           "name": "Long face"
          },
          {
           "hpo": "HP:0000006",
           "id": "OMIM:608930",
           "name": "Autosomal dominant inheritance"
          },
          {
           "hpo": "HP:0008872",
           "id": "OMIM:605809",
           "name": "Feeding difficulties in infancy"
          },
          {
           "hpo": "HP:0001765",
           "id": "OMIM:616040",
           "name": "Hammertoe"
          },
          {
           "hpo": "HP:0003443",
           "id": "OMIM:601462",
           "name": "Decreased size of nerve terminals"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616326",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:605809",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:608930",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0003402",
           "id": "OMIM:254210",
           "name": "Decreased miniature endplate potentials"
          },
          {
           "hpo": "HP:0001284",
           "id": "OMIM:616040",
           "name": "Areflexia"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:615120",
           "name": "High palate"
          },
          {
           "hpo": "HP:0002715",
           "id": "OMIM:254210",
           "name": "Abnormality of the immune system"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:254300",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:601462",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:616322",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0003828",
           "id": "OMIM:615120",
           "name": "Variable expressivity"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:615120",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0003324",
           "id": "OMIM:608930",
           "name": "Generalized muscle weakness"
          },
          {
           "hpo": "HP:0001265",
           "id": "OMIM:616304",
           "name": "Hyporeflexia"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:615120",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:254210",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:608931",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0001283",
           "id": "OMIM:608930",
           "name": "Bulbar palsy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616321",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0003403",
           "id": "OMIM:605809",
           "name": "EMG: decremental response of compound muscle action potential to repetitive nerve stimulation"
          },
          {
           "hpo": "HP:0002465",
           "id": "OMIM:616330",
           "name": "Poor speech"
          },
          {
           "hpo": "HP:0000400",
           "id": "OMIM:608930",
           "name": "Macrotia"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:616313",
           "name": "High palate"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:616321",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:608930",
           "name": "High palate"
          },
          {
           "hpo": "HP:0000275",
           "id": "OMIM:616313",
           "name": "Narrow face"
          },
          {
           "hpo": "HP:0000486",
           "id": "OMIM:254210",
           "name": "Strabismus"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:608930",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616330",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:605809",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0002715",
           "id": "OMIM:254300",
           "name": "Abnormality of the immune system"
          },
          {
           "hpo": "HP:0001265",
           "id": "OMIM:616040",
           "name": "Hyporeflexia"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616324",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:608931",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0000467",
           "id": "OMIM:616322",
           "name": "Neck muscle weakness"
          },
          {
           "hpo": "HP:0003402",
           "id": "OMIM:605809",
           "name": "Decreased miniature endplate potentials"
          },
          {
           "hpo": "HP:0001284",
           "id": "OMIM:616330",
           "name": "Areflexia"
          },
          {
           "hpo": "HP:0003394",
           "id": "OMIM:254300",
           "name": "Muscle cramps"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:254300",
           "name": "Facial palsy"
          }
         ],
         "hgvs": [
          "ENST00000379370(ENSG00000188157):c.1178-6T>C"
         ],
         "populationFrequencies": [
          {
           "altAlleleFreq": 0.9662577,
           "population": "SAS",
           "refAlleleFreq": 0.03374231,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.71686745,
           "population": "AFR",
           "refAlleleFreq": 0.28313255,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.68986386,
           "population": "AFR",
           "refAlleleFreq": 0.31013614,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 1,
           "population": "JPT",
           "refAlleleFreq": 0,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.6944444,
           "population": "YRI",
           "refAlleleFreq": 0.30555558,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.921317,
           "population": "ALL",
           "refAlleleFreq": 0.07868302,
           "study": "DISCOVEHR"
          },
          {
           "altAlleleFreq": 0.93434346,
           "population": "CEU",
           "refAlleleFreq": 0.06565654,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9207771,
           "population": "EA",
           "refAlleleFreq": 0.07922292,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.9165589,
           "population": "ALL",
           "refAlleleFreq": 0.08344108,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.99976623,
           "population": "EAS",
           "refAlleleFreq": 0.00023376942,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.99900794,
           "population": "EAS",
           "refAlleleFreq": 0.0009920597,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9325581,
           "population": "OTH",
           "refAlleleFreq": 0.06744188,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.92052364,
           "population": "ALL",
           "refAlleleFreq": 0.07947636,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.9198803,
           "population": "FEMALE",
           "refAlleleFreq": 0.08011973,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.9160947,
           "population": "NFE",
           "refAlleleFreq": 0.08390528,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.7194457,
           "population": "AA",
           "refAlleleFreq": 0.2805543,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.922216,
           "population": "TWINSUK_NODUP",
           "refAlleleFreq": 0.077784,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 1,
           "population": "EAS",
           "refAlleleFreq": 0,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.91904515,
           "population": "ALSPAC",
           "refAlleleFreq": 0.08095485,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.74590164,
           "population": "ASW",
           "refAlleleFreq": 0.25409836,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7352941,
           "population": "MSL",
           "refAlleleFreq": 0.2647059,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.95126826,
           "population": "AMR",
           "refAlleleFreq": 0.048731744,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.9444444,
           "population": "FIN",
           "refAlleleFreq": 0.055555582,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9997642,
           "population": "EAS",
           "refAlleleFreq": 0.00023579597,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.88461536,
           "population": "PUR",
           "refAlleleFreq": 0.11538464,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.93301433,
           "population": "AMR",
           "refAlleleFreq": 0.06698567,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.9519038,
           "population": "ALL",
           "refAlleleFreq": 0.04809618,
           "study": "GONL"
          },
          {
           "altAlleleFreq": 0.7079646,
           "population": "GWD",
           "refAlleleFreq": 0.2920354,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.93406594,
           "population": "GBR",
           "refAlleleFreq": 0.06593406,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.91588783,
           "population": "IBS",
           "refAlleleFreq": 0.08411217,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.95406914,
           "population": "AMR",
           "refAlleleFreq": 0.045930862,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.98255813,
           "population": "BEB",
           "refAlleleFreq": 0.017441869,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.6510417,
           "population": "ACB",
           "refAlleleFreq": 0.3489583,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7171717,
           "population": "ESN",
           "refAlleleFreq": 0.28282827,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.91767037,
           "population": "NFE",
           "refAlleleFreq": 0.08232963,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.90957445,
           "population": "CLM",
           "refAlleleFreq": 0.09042555,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9347826,
           "population": "OTH",
           "refAlleleFreq": 0.065217376,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.879539,
           "population": "ALL",
           "refAlleleFreq": 0.12046099,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.95098037,
           "population": "ITU",
           "refAlleleFreq": 0.049019635,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.92316216,
           "population": "OTH",
           "refAlleleFreq": 0.07683784,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.93297297,
           "population": "NFE",
           "refAlleleFreq": 0.06702703,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.9234955,
           "population": "ALL",
           "refAlleleFreq": 0.07650453,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 1,
           "population": "CHS",
           "refAlleleFreq": 0,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9479167,
           "population": "PJL",
           "refAlleleFreq": 0.052083313,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.6142322,
           "population": "ALL",
           "refAlleleFreq": 0.38576782,
           "study": "MGP"
          },
          {
           "altAlleleFreq": 0.95436895,
           "population": "SAS",
           "refAlleleFreq": 0.04563105,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.9647059,
           "population": "PEL",
           "refAlleleFreq": 0.035294116,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 1,
           "population": "CHB",
           "refAlleleFreq": 0,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.71750665,
           "population": "AFR",
           "refAlleleFreq": 0.28249335,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.8851837,
           "population": "ALL",
           "refAlleleFreq": 0.11481631,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9375,
           "population": "MXL",
           "refAlleleFreq": 0.0625,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9542195,
           "population": "FIN",
           "refAlleleFreq": 0.04578048,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.9207493,
           "population": "AMR",
           "refAlleleFreq": 0.07925069,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9611651,
           "population": "GIH",
           "refAlleleFreq": 0.03883493,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.89719623,
           "population": "TSI",
           "refAlleleFreq": 0.10280377,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8525927,
           "population": "ALL",
           "refAlleleFreq": 0.1474073,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.92206043,
           "population": "TWINSUK",
           "refAlleleFreq": 0.07793957,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.95972383,
           "population": "FIN",
           "refAlleleFreq": 0.04027617,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.99019605,
           "population": "STU",
           "refAlleleFreq": 0.009803951,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 1,
           "population": "CDX",
           "refAlleleFreq": 0,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.92441624,
           "population": "ASJ",
           "refAlleleFreq": 0.075583756,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.9264251,
           "population": "MALE",
           "refAlleleFreq": 0.0735749,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.8816329,
           "population": "FEMALE",
           "refAlleleFreq": 0.118367076,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.9949495,
           "population": "KHV",
           "refAlleleFreq": 0.0050504804,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8778399,
           "population": "MALE",
           "refAlleleFreq": 0.12216008,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.93,
           "population": "ASJ",
           "refAlleleFreq": 0.06999999,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.6010101,
           "population": "LWK",
           "refAlleleFreq": 0.39898992,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.70930856,
           "population": "AFR",
           "refAlleleFreq": 0.29069144,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.94924814,
           "population": "FIN",
           "refAlleleFreq": 0.050751865,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.92445326,
           "population": "EUR",
           "refAlleleFreq": 0.07554674,
           "study": "1kG_phase3"
          }
         ],
         "reference": "T",
         "repeat": [],
         "start": 977330,
         "traitAssociation": [
          {
           "additionalProperties": [
            {
             "name": "ClinicalSignificance_in_source_file",
             "value": "benign"
            }
           ],
           "bibliography": [],
           "ethnicity": "Z",
           "heritableTraits": [
            {
             "inheritanceMode": "biallelic",
             "trait": "not specified"
            }
           ],
           "id": "RCV000116254",
           "source": {
            "name": "clinvar"
           },
           "submissions": [],
           "url": "https://www.ncbi.nlm.nih.gov/clinvar/RCV000116254",
           "variantClassification": {
            "clinicalSignificance": "benign"
           }
          },
          {
           "additionalProperties": [
            {
             "name": "ClinicalSignificance_in_source_file",
             "value": "benign"
            }
           ],
           "bibliography": [],
           "ethnicity": "Z",
           "heritableTraits": [
            {
             "inheritanceMode": "unknown",
             "trait": "not specified"
            }
           ],
           "id": "128292",
           "source": {
            "name": "clinvar"
           },
           "submissions": [],
           "variantClassification": {
            "clinicalSignificance": "benign"
           }
          }
         ],
         "variantTraitAssociation": {
          "clinvar": [
           {
            "accession": "RCV000116254",
            "clinicalSignificance": "benign",
            "geneNames": [],
            "reviewStatus": "",
            "traits": [
             "{\"trait\": \"not specified\", \"inheritanceMode\": \"biallelic\"}"
            ]
           },
           {
            "accession": "128292",
            "clinicalSignificance": "benign",
            "geneNames": [],
            "reviewStatus": "",
            "traits": [
             "{\"trait\": \"not specified\", \"inheritanceMode\": \"unknown\"}"
            ]
           }
          ]
         },
         "xrefs": [
          {
           "id": "ENSG00000188157",
           "source": "ensemblGene"
          },
          {
           "id": "ENST00000479707",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000379370",
           "source": "ensemblTranscript"
          },
          {
           "id": "rs2799066",
           "source": "dbSNP"
          },
          {
           "id": "ENST00000477585",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000469403",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000379370(ENSG00000188157):c.1178-6T>C",
           "source": "ensemblTranscript"
          }
         ]
        },
        "chromosome": "1",
        "end": 977330,
        "id": "1:977330:T:C",
        "length": 1,
        "names": [],
        "reference": "T",
        "start": 977330,
        "strand": "+",
        "studies": [],
        "type": "SNV"
       },
       {
        "alternate": "T",
        "annotation": {
         "alternate": "T",
         "chromosome": "1",
         "consequenceTypes": [
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000419249",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "mRNA_end_NF",
            "cds_end_NF",
            "cds_start_NF",
            "mRNA_start_NF"
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000492947",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001632",
             "name": "downstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000379370",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "sense_intronic",
           "ensemblGeneId": "ENSG00000242590",
           "ensemblTranscriptId": "ENST00000418300",
           "geneName": "RP11-54O7.14",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "basic"
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000466223",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001632",
             "name": "downstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000478677",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001632",
             "name": "downstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000188157",
           "ensemblTranscriptId": "ENST00000461111",
           "geneName": "AGRN",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ]
          },
          {
           "ensemblGeneId": "",
           "ensemblTranscriptId": "",
           "geneName": "",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001566",
             "name": "regulatory_region_variant"
            }
           ]
          }
         ],
         "conservation": [
          {
           "description": "",
           "score": -1.225000023841858,
           "source": "phylop"
          },
          {
           "description": "",
           "score": 0.020999999716877937,
           "source": "phastCons"
          },
          {
           "description": "",
           "score": -6.110000133514404,
           "source": "gerp"
          }
         ],
         "cytoband": [
          {
           "chromosome": "1",
           "end": 2300000,
           "name": "p36.33",
           "stain": "gneg",
           "start": 1
          }
         ],
         "displayConsequenceType": "splice_region_variant",
         "end": 987200,
         "functionalScore": [
          {
           "description": "",
           "score": 0.26000022888183594,
           "source": "cadd_raw"
          },
          {
           "description": "",
           "score": 5.349999904632568,
           "source": "cadd_scaled"
          }
         ],
         "geneTraitAssociation": [
          {
           "hpo": "HP:0003388",
           "id": "OMIM:614198",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616321",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:254300",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0003402",
           "id": "OMIM:608930",
           "name": "Decreased miniature endplate potentials"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:254210",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:601462",
           "name": "High palate"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:608930",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:254300",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0003443",
           "id": "OMIM:608931",
           "name": "Decreased size of nerve terminals"
          },
          {
           "hpo": "HP:0000276",
           "id": "OMIM:608931",
           "name": "Long face"
          },
          {
           "hpo": "HP:0002872",
           "id": "OMIM:254210",
           "name": "Apneic episodes precipitated by illness, fatigue, stress"
          },
          {
           "hpo": "HP:0001260",
           "id": "OMIM:601462",
           "name": "Dysarthria"
          },
          {
           "hpo": "HP:0001263",
           "id": "OMIM:616330",
           "name": "Global developmental delay"
          },
          {
           "hpo": "HP:0002033",
           "id": "OMIM:254210",
           "name": "Poor suck"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:608931",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0001270",
           "id": "OMIM:608931",
           "name": "Motor delay"
          },
          {
           "hpo": "HP:0008180",
           "id": "OMIM:254300",
           "name": "Mildly elevated creatine phosphokinase"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:608931",
           "name": "High palate"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:608930",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0001260",
           "id": "OMIM:608930",
           "name": "Dysarthria"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:605809",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:608931",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0002804",
           "id": "OMIM:254210",
           "name": "Arthrogryposis multiplex congenita"
          },
          {
           "hpo": "HP:0001283",
           "id": "OMIM:605809",
           "name": "Bulbar palsy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:605809",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0000276",
           "id": "OMIM:616313",
           "name": "Long face"
          },
          {
           "hpo": "HP:0002515",
           "id": "OMIM:254300",
           "name": "Waddling gait"
          },
          {
           "hpo": "HP:0001612",
           "id": "OMIM:254210",
           "name": "Weak cry"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:616323",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0001270",
           "id": "OMIM:614198",
           "name": "Motor delay"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:616324",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616323",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000544",
           "id": "OMIM:614198",
           "name": "External ophthalmoplegia"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:605809",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:601462",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0003403",
           "id": "OMIM:608931",
           "name": "EMG: decremental response of compound muscle action potential to repetitive nerve stimulation"
          },
          {
           "hpo": "HP:0000486",
           "id": "OMIM:608931",
           "name": "Strabismus"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:605809",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:608931",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0003828",
           "id": "OMIM:608931",
           "name": "Variable expressivity"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:616322",
           "name": "High palate"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:608931",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616326",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:616326",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0001283",
           "id": "OMIM:254210",
           "name": "Bulbar palsy"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616323",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:601462",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0003593",
           "id": "OMIM:608930",
           "name": "Infantile onset"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:608930",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0001288",
           "id": "OMIM:616040",
           "name": "Gait disturbance"
          },
          {
           "hpo": "HP:0001260",
           "id": "OMIM:608931",
           "name": "Dysarthria"
          },
          {
           "hpo": "HP:0001270",
           "id": "OMIM:608930",
           "name": "Motor delay"
          },
          {
           "hpo": "HP:0003324",
           "id": "OMIM:601462",
           "name": "Generalized muscle weakness"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:605809",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616325",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616323",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:614198",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002098",
           "id": "OMIM:605809",
           "name": "Respiratory distress"
          },
          {
           "hpo": "HP:0001612",
           "id": "OMIM:605809",
           "name": "Weak cry"
          },
          {
           "hpo": "HP:0002715",
           "id": "OMIM:608931",
           "name": "Abnormality of the immune system"
          },
          {
           "hpo": "HP:0003693",
           "id": "OMIM:254300",
           "name": "Distal amyotrophy"
          },
          {
           "hpo": "HP:0000467",
           "id": "OMIM:616325",
           "name": "Neck muscle weakness"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616304",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:614198",
           "name": "High palate"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:608931",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:608931",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0007126",
           "id": "OMIM:254300",
           "name": "Proximal amyotrophy"
          },
          {
           "hpo": "HP:0000467",
           "id": "OMIM:608930",
           "name": "Neck muscle weakness"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:254210",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:605809",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616314",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:254210",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616326",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0000303",
           "id": "OMIM:608931",
           "name": "Mandibular prognathia"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616326",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:616326",
           "name": "High palate"
          },
          {
           "hpo": "HP:0003202",
           "id": "OMIM:616040",
           "name": "Skeletal muscle atrophy"
          },
          {
           "hpo": "HP:0001324",
           "id": "OMIM:616314",
           "name": "Muscle weakness"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:616323",
           "name": "High palate"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:608930",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0001761",
           "id": "OMIM:616040",
           "name": "Pes cavus"
          },
          {
           "hpo": "HP:0003394",
           "id": "OMIM:608931",
           "name": "Muscle cramps"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616325",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:254300",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616322",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616325",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616330",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616304",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002882",
           "id": "OMIM:254210",
           "name": "Sudden episodic apnea"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616313",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:608931",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0002650",
           "id": "OMIM:608930",
           "name": "Scoliosis"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:608931",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0003202",
           "id": "OMIM:616313",
           "name": "Skeletal muscle atrophy"
          },
          {
           "hpo": "HP:0001324",
           "id": "OMIM:616330",
           "name": "Muscle weakness"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:616326",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0000347",
           "id": "OMIM:608930",
           "name": "Micrognathia"
          },
          {
           "hpo": "HP:0003593",
           "id": "OMIM:608931",
           "name": "Infantile onset"
          },
          {
           "hpo": "HP:0003621",
           "id": "OMIM:254300",
           "name": "Juvenile onset"
          },
          {
           "hpo": "HP:0001371",
           "id": "OMIM:616313",
           "name": "Flexion contracture"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:614198",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:615120",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001251",
           "id": "OMIM:616330",
           "name": "Ataxia"
          },
          {
           "hpo": "HP:0002804",
           "id": "OMIM:608930",
           "name": "Arthrogryposis multiplex congenita"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616324",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:601462",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0001283",
           "id": "OMIM:254300",
           "name": "Bulbar palsy"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:254210",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:608930",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0002098",
           "id": "OMIM:254210",
           "name": "Respiratory distress"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616313",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616322",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0000486",
           "id": "OMIM:605809",
           "name": "Strabismus"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616325",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:601462",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0003403",
           "id": "OMIM:254210",
           "name": "EMG: decremental response of compound muscle action potential to repetitive nerve stimulation"
          },
          {
           "hpo": "HP:0003674",
           "id": "OMIM:601462",
           "name": "Onset"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616324",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0003397",
           "id": "OMIM:605809",
           "name": "Generalized hypotonia due to defect at the neuromuscular junction"
          },
          {
           "hpo": "HP:0002355",
           "id": "OMIM:616304",
           "name": "Difficulty walking"
          },
          {
           "hpo": "HP:0003397",
           "id": "OMIM:254210",
           "name": "Generalized hypotonia due to defect at the neuromuscular junction"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616322",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616322",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616325",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:601462",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0001371",
           "id": "OMIM:616330",
           "name": "Flexion contracture"
          },
          {
           "hpo": "HP:0002804",
           "id": "OMIM:608931",
           "name": "Arthrogryposis multiplex congenita"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:254300",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0003199",
           "id": "OMIM:608931",
           "name": "Decreased muscle mass"
          },
          {
           "hpo": "HP:0000365",
           "id": "OMIM:616040",
           "name": "Hearing impairment"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616322",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:608930",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:254210",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0000467",
           "id": "OMIM:616324",
           "name": "Neck muscle weakness"
          },
          {
           "hpo": "HP:0003577",
           "id": "OMIM:254210",
           "name": "Congenital onset"
          },
          {
           "hpo": "HP:0003828",
           "id": "OMIM:608930",
           "name": "Variable expressivity"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:254300",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:614198",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:616325",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:616322",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616330",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616323",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:608931",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:616324",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:616325",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:615120",
           "name": "Facial palsy"
          },
          {
           "hpo": "HP:0001260",
           "id": "OMIM:616330",
           "name": "Dysarthria"
          },
          {
           "hpo": "HP:0002033",
           "id": "OMIM:608930",
           "name": "Poor suck"
          },
          {
           "hpo": "HP:0003690",
           "id": "OMIM:605809",
           "name": "Limb muscle weakness"
          },
          {
           "hpo": "HP:0001612",
           "id": "OMIM:608930",
           "name": "Weak cry"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616322",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0003436",
           "id": "OMIM:601462",
           "name": "Prolonged miniature endplate currents"
          },
          {
           "hpo": "HP:0001612",
           "id": "OMIM:608931",
           "name": "Weak cry"
          },
          {
           "hpo": "HP:0000006",
           "id": "OMIM:601462",
           "name": "Autosomal dominant inheritance"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616324",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616324",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0003701",
           "id": "OMIM:616040",
           "name": "Proximal muscle weakness"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616321",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:254300",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0004889",
           "id": "OMIM:601462",
           "name": "Intermittent episodes of respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0003577",
           "id": "OMIM:605809",
           "name": "Congenital onset"
          },
          {
           "hpo": "HP:0001290",
           "id": "OMIM:605809",
           "name": "Generalized hypotonia"
          },
          {
           "hpo": "HP:0003680",
           "id": "OMIM:608931",
           "name": "Nonprogressive"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616313",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002033",
           "id": "OMIM:605809",
           "name": "Poor suck"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:254300",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0003324",
           "id": "OMIM:616321",
           "name": "Generalized muscle weakness"
          },
          {
           "hpo": "HP:0000689",
           "id": "OMIM:608931",
           "name": "Dental malocclusion"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:615120",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0003202",
           "id": "OMIM:608931",
           "name": "Skeletal muscle atrophy"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:616321",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002104",
           "id": "OMIM:614198",
           "name": "Apnea"
          },
          {
           "hpo": "HP:0001288",
           "id": "OMIM:614198",
           "name": "Gait disturbance"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:608931",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:608931",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0002804",
           "id": "OMIM:616326",
           "name": "Arthrogryposis multiplex congenita"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616323",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616326",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0002872",
           "id": "OMIM:605809",
           "name": "Apneic episodes precipitated by illness, fatigue, stress"
          },
          {
           "hpo": "HP:0003554",
           "id": "OMIM:608930",
           "name": "Type 2 muscle fiber atrophy"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:254210",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:616314",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0002355",
           "id": "OMIM:616330",
           "name": "Difficulty walking"
          },
          {
           "hpo": "HP:0003307",
           "id": "OMIM:614198",
           "name": "Hyperlordosis"
          },
          {
           "hpo": "HP:0001558",
           "id": "OMIM:616330",
           "name": "Decreased fetal movement"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616314",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:616313",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0000007",
           "id": "OMIM:608930",
           "name": "Autosomal recessive inheritance"
          },
          {
           "hpo": "HP:0002882",
           "id": "OMIM:605809",
           "name": "Sudden episodic apnea"
          },
          {
           "hpo": "HP:0000508",
           "id": "OMIM:601462",
           "name": "Ptosis"
          },
          {
           "hpo": "HP:0002540",
           "id": "OMIM:616321",
           "name": "Inability to walk"
          },
          {
           "hpo": "HP:0000774",
           "id": "OMIM:615120",
           "name": "Narrow chest"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616313",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0000276",
           "id": "OMIM:616326",
           "name": "Long face"
          },
          {
           "hpo": "HP:0000006",
           "id": "OMIM:608930",
           "name": "Autosomal dominant inheritance"
          },
          {
           "hpo": "HP:0008872",
           "id": "OMIM:605809",
           "name": "Feeding difficulties in infancy"
          },
          {
           "hpo": "HP:0001765",
           "id": "OMIM:616040",
           "name": "Hammertoe"
          },
          {
           "hpo": "HP:0003443",
           "id": "OMIM:601462",
           "name": "Decreased size of nerve terminals"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:616326",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0002747",
           "id": "OMIM:605809",
           "name": "Respiratory insufficiency due to muscle weakness"
          },
          {
           "hpo": "HP:0003391",
           "id": "OMIM:608930",
           "name": "Gowers sign"
          },
          {
           "hpo": "HP:0003402",
           "id": "OMIM:254210",
           "name": "Decreased miniature endplate potentials"
          },
          {
           "hpo": "HP:0001284",
           "id": "OMIM:616040",
           "name": "Areflexia"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:615120",
           "name": "High palate"
          },
          {
           "hpo": "HP:0002715",
           "id": "OMIM:254210",
           "name": "Abnormality of the immune system"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:254300",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0011968",
           "id": "OMIM:601462",
           "name": "Feeding difficulties"
          },
          {
           "hpo": "HP:0000602",
           "id": "OMIM:616322",
           "name": "Ophthalmoplegia"
          },
          {
           "hpo": "HP:0003828",
           "id": "OMIM:615120",
           "name": "Variable expressivity"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:615120",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0003324",
           "id": "OMIM:608930",
           "name": "Generalized muscle weakness"
          },
          {
           "hpo": "HP:0001265",
           "id": "OMIM:616304",
           "name": "Hyporeflexia"
          },
          {
           "hpo": "HP:0002093",
           "id": "OMIM:615120",
           "name": "Respiratory insufficiency"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:254210",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:608931",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0001283",
           "id": "OMIM:608930",
           "name": "Bulbar palsy"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616321",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0003403",
           "id": "OMIM:605809",
           "name": "EMG: decremental response of compound muscle action potential to repetitive nerve stimulation"
          },
          {
           "hpo": "HP:0002465",
           "id": "OMIM:616330",
           "name": "Poor speech"
          },
          {
           "hpo": "HP:0000400",
           "id": "OMIM:608930",
           "name": "Macrotia"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:616313",
           "name": "High palate"
          },
          {
           "hpo": "HP:0002015",
           "id": "OMIM:616321",
           "name": "Dysphagia"
          },
          {
           "hpo": "HP:0000218",
           "id": "OMIM:608930",
           "name": "High palate"
          },
          {
           "hpo": "HP:0000275",
           "id": "OMIM:616313",
           "name": "Narrow face"
          },
          {
           "hpo": "HP:0000486",
           "id": "OMIM:254210",
           "name": "Strabismus"
          },
          {
           "hpo": "HP:0000597",
           "id": "OMIM:608930",
           "name": "Ophthalmoparesis"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:616330",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0003473",
           "id": "OMIM:605809",
           "name": "Fatigable weakness"
          },
          {
           "hpo": "HP:0002715",
           "id": "OMIM:254300",
           "name": "Abnormality of the immune system"
          },
          {
           "hpo": "HP:0001265",
           "id": "OMIM:616040",
           "name": "Hyporeflexia"
          },
          {
           "hpo": "HP:0001252",
           "id": "OMIM:616324",
           "name": "Muscular hypotonia"
          },
          {
           "hpo": "HP:0003388",
           "id": "OMIM:608931",
           "name": "Easy fatigability"
          },
          {
           "hpo": "HP:0000467",
           "id": "OMIM:616322",
           "name": "Neck muscle weakness"
          },
          {
           "hpo": "HP:0003402",
           "id": "OMIM:605809",
           "name": "Decreased miniature endplate potentials"
          },
          {
           "hpo": "HP:0001284",
           "id": "OMIM:616330",
           "name": "Areflexia"
          },
          {
           "hpo": "HP:0003394",
           "id": "OMIM:254300",
           "name": "Muscle cramps"
          },
          {
           "hpo": "HP:0010628",
           "id": "OMIM:254300",
           "name": "Facial palsy"
          }
         ],
         "hgvs": [
          "ENST00000379370(ENSG00000188157):c.5651+5C>T",
          "ENST00000419249(ENSG00000188157):c.558+5C>T"
         ],
         "populationFrequencies": [
          {
           "altAlleleFreq": 0.9018405,
           "population": "SAS",
           "refAlleleFreq": 0.09815949,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.45132336,
           "population": "AFR",
           "refAlleleFreq": 0.5486766,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.40166414,
           "population": "AFR",
           "refAlleleFreq": 0.59833586,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 1,
           "population": "JPT",
           "refAlleleFreq": 0,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.36574075,
           "population": "YRI",
           "refAlleleFreq": 0.6342592,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.906793,
           "population": "ALL",
           "refAlleleFreq": 0.093207,
           "study": "DISCOVEHR"
          },
          {
           "altAlleleFreq": 0.9292929,
           "population": "CEU",
           "refAlleleFreq": 0.07070708,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.90997905,
           "population": "EA",
           "refAlleleFreq": 0.090020955,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.8756909,
           "population": "ALL",
           "refAlleleFreq": 0.12430912,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.99570715,
           "population": "EAS",
           "refAlleleFreq": 0.0042928457,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.9930556,
           "population": "EAS",
           "refAlleleFreq": 0.006944418,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9101124,
           "population": "OTH",
           "refAlleleFreq": 0.08988762,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.9100767,
           "population": "ALL",
           "refAlleleFreq": 0.08992332,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.88273424,
           "population": "FEMALE",
           "refAlleleFreq": 0.11726576,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.905529,
           "population": "NFE",
           "refAlleleFreq": 0.09447098,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.45797366,
           "population": "AA",
           "refAlleleFreq": 0.54202634,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.9135423,
           "population": "TWINSUK_NODUP",
           "refAlleleFreq": 0.08645773,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.99691355,
           "population": "EAS",
           "refAlleleFreq": 0.0030864477,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.9076284,
           "population": "ALSPAC",
           "refAlleleFreq": 0.09237158,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.5081967,
           "population": "ASW",
           "refAlleleFreq": 0.4918033,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.42941177,
           "population": "MSL",
           "refAlleleFreq": 0.57058823,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9322327,
           "population": "AMR",
           "refAlleleFreq": 0.06776732,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.93434346,
           "population": "FIN",
           "refAlleleFreq": 0.06565654,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9967585,
           "population": "EAS",
           "refAlleleFreq": 0.0032414794,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.83653843,
           "population": "PUR",
           "refAlleleFreq": 0.16346157,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.92124104,
           "population": "AMR",
           "refAlleleFreq": 0.078758955,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.94388777,
           "population": "ALL",
           "refAlleleFreq": 0.05611223,
           "study": "GONL"
          },
          {
           "altAlleleFreq": 0.45132744,
           "population": "GWD",
           "refAlleleFreq": 0.54867256,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9230769,
           "population": "GBR",
           "refAlleleFreq": 0.07692307,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9205608,
           "population": "IBS",
           "refAlleleFreq": 0.07943922,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.93660015,
           "population": "AMR",
           "refAlleleFreq": 0.06339985,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.8895349,
           "population": "BEB",
           "refAlleleFreq": 0.11046511,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.40104166,
           "population": "ACB",
           "refAlleleFreq": 0.5989584,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.3939394,
           "population": "ESN",
           "refAlleleFreq": 0.6060606,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.90784025,
           "population": "NFE",
           "refAlleleFreq": 0.09215975,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.86702126,
           "population": "CLM",
           "refAlleleFreq": 0.13297874,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.905102,
           "population": "OTH",
           "refAlleleFreq": 0.094897985,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.7970198,
           "population": "ALL",
           "refAlleleFreq": 0.20298022,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.88235295,
           "population": "ITU",
           "refAlleleFreq": 0.11764705,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.89622986,
           "population": "OTH",
           "refAlleleFreq": 0.10377014,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.9236891,
           "population": "NFE",
           "refAlleleFreq": 0.07631087,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.8880143,
           "population": "ALL",
           "refAlleleFreq": 0.11198568,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.9904762,
           "population": "CHS",
           "refAlleleFreq": 0.009523809,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8802083,
           "population": "PJL",
           "refAlleleFreq": 0.11979169,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8951311,
           "population": "ALL",
           "refAlleleFreq": 0.10486889,
           "study": "MGP"
          },
          {
           "altAlleleFreq": 0.8844428,
           "population": "SAS",
           "refAlleleFreq": 0.115557194,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.9411765,
           "population": "PEL",
           "refAlleleFreq": 0.058823526,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 1,
           "population": "CHB",
           "refAlleleFreq": 0,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.45202267,
           "population": "AFR",
           "refAlleleFreq": 0.5479773,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.788738,
           "population": "ALL",
           "refAlleleFreq": 0.21126199,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9296875,
           "population": "MXL",
           "refAlleleFreq": 0.0703125,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9504585,
           "population": "FIN",
           "refAlleleFreq": 0.049541473,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.88760805,
           "population": "AMR",
           "refAlleleFreq": 0.11239195,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9368932,
           "population": "GIH",
           "refAlleleFreq": 0.063106775,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.86915886,
           "population": "TSI",
           "refAlleleFreq": 0.13084114,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7569231,
           "population": "ALL",
           "refAlleleFreq": 0.24307692,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.9126214,
           "population": "TWINSUK",
           "refAlleleFreq": 0.08737862,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.95412844,
           "population": "FIN",
           "refAlleleFreq": 0.045871556,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.9166667,
           "population": "STU",
           "refAlleleFreq": 0.08333331,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.97849464,
           "population": "CDX",
           "refAlleleFreq": 0.021505356,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8919194,
           "population": "ASJ",
           "refAlleleFreq": 0.108080626,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.89235705,
           "population": "MALE",
           "refAlleleFreq": 0.10764295,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.805302,
           "population": "FEMALE",
           "refAlleleFreq": 0.19469798,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.9949495,
           "population": "KHV",
           "refAlleleFreq": 0.0050504804,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7903188,
           "population": "MALE",
           "refAlleleFreq": 0.20968121,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.8907285,
           "population": "ASJ",
           "refAlleleFreq": 0.10927153,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.3030303,
           "population": "LWK",
           "refAlleleFreq": 0.6969697,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.43764782,
           "population": "AFR",
           "refAlleleFreq": 0.5623522,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.94429183,
           "population": "FIN",
           "refAlleleFreq": 0.05570817,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.91451293,
           "population": "EUR",
           "refAlleleFreq": 0.08548707,
           "study": "1kG_phase3"
          }
         ],
         "reference": "C",
         "repeat": [],
         "start": 987200,
         "traitAssociation": [
          {
           "additionalProperties": [
            {
             "name": "ClinicalSignificance_in_source_file",
             "value": "benign"
            }
           ],
           "bibliography": [],
           "ethnicity": "Z",
           "heritableTraits": [
            {
             "inheritanceMode": "biallelic",
             "trait": "not specified"
            }
           ],
           "id": "RCV000116279",
           "source": {
            "name": "clinvar"
           },
           "submissions": [],
           "url": "https://www.ncbi.nlm.nih.gov/clinvar/RCV000116279",
           "variantClassification": {
            "clinicalSignificance": "benign"
           }
          },
          {
           "additionalProperties": [
            {
             "name": "ClinicalSignificance_in_source_file",
             "value": "benign"
            }
           ],
           "bibliography": [],
           "ethnicity": "Z",
           "heritableTraits": [
            {
             "inheritanceMode": "unknown",
             "trait": "not specified"
            }
           ],
           "id": "128317",
           "source": {
            "name": "clinvar"
           },
           "submissions": [],
           "variantClassification": {
            "clinicalSignificance": "benign"
           }
          }
         ],
         "variantTraitAssociation": {
          "clinvar": [
           {
            "accession": "RCV000116279",
            "clinicalSignificance": "benign",
            "geneNames": [],
            "reviewStatus": "",
            "traits": [
             "{\"trait\": \"not specified\", \"inheritanceMode\": \"biallelic\"}"
            ]
           },
           {
            "accession": "128317",
            "clinicalSignificance": "benign",
            "geneNames": [],
            "reviewStatus": "",
            "traits": [
             "{\"trait\": \"not specified\", \"inheritanceMode\": \"unknown\"}"
            ]
           }
          ]
         },
         "xrefs": [
          {
           "id": "rs9803031",
           "source": "dbSNP"
          },
          {
           "id": "ENST00000419249",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000492947",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000419249(ENSG00000188157):c.558+5C>T",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000466223",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000461111",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENSG00000242590",
           "source": "ensemblGene"
          },
          {
           "id": "ENSG00000188157",
           "source": "ensemblGene"
          },
          {
           "id": "ENST00000379370",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000418300",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000379370(ENSG00000188157):c.5651+5C>T",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000478677",
           "source": "ensemblTranscript"
          }
         ]
        },
        "chromosome": "1",
        "end": 987200,
        "id": "1:987200:C:T",
        "length": 1,
        "names": [],
        "reference": "C",
        "start": 987200,
        "strand": "+",
        "studies": [],
        "type": "SNV"
       },
       {
        "alternate": "C",
        "annotation": {
         "alternate": "C",
         "chromosome": "1",
         "consequenceTypes": [
          {
           "biotype": "processed_transcript",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000379037",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "mRNA_end_NF",
            "mRNA_start_NF"
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000353662",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "basic"
           ]
          },
          {
           "biotype": "nonsense_mediated_decay",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000354980",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0002083",
             "name": "2KB_downstream_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000470659",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "processed_transcript",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000438966",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001632",
             "name": "downstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000492936",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001619",
             "name": "non_coding_transcript_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000354700",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000467278",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001619",
             "name": "non_coding_transcript_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000479108",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001632",
             "name": "downstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000476572",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001630",
             "name": "splice_region_variant"
            },
            {
             "accession": "SO:0001619",
             "name": "non_coding_transcript_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000493992",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000478065",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001632",
             "name": "downstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000131584",
           "ensemblTranscriptId": "ENST00000472541",
           "geneName": "ACAP3",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0002083",
             "name": "2KB_downstream_variant"
            }
           ]
          },
          {
           "ensemblGeneId": "",
           "ensemblTranscriptId": "",
           "geneName": "",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001566",
             "name": "regulatory_region_variant"
            }
           ]
          }
         ],
         "conservation": [
          {
           "description": "",
           "score": -0.40400001406669617,
           "source": "phylop"
          },
          {
           "description": "",
           "score": 0.12399999797344208,
           "source": "phastCons"
          },
          {
           "description": "",
           "score": 0.9129999876022339,
           "source": "gerp"
          }
         ],
         "cytoband": [
          {
           "chromosome": "1",
           "end": 2300000,
           "name": "p36.33",
           "stain": "gneg",
           "start": 1
          }
         ],
         "displayConsequenceType": "splice_region_variant",
         "end": 1233941,
         "functionalScore": [
          {
           "description": "",
           "score": -0.22999954223632812,
           "source": "cadd_raw"
          },
          {
           "description": "",
           "score": 0.949999988079071,
           "source": "cadd_scaled"
          }
         ],
         "hgvs": [
          "ENST00000354700(ENSG00000131584):c.863+6A>G",
          "ENST00000353662(ENSG00000131584):c.737+6A>G",
          "ENST00000492936(ENSG00000131584):n.2602+6A>G",
          "ENST00000467278(ENSG00000131584):n.389+6A>G",
          "ENST00000476572(ENSG00000131584):n.386+6A>G"
         ],
         "populationFrequencies": [
          {
           "altAlleleFreq": 0.38036808,
           "population": "SAS",
           "refAlleleFreq": 0.6196319,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7274893,
           "population": "AFR",
           "refAlleleFreq": 0.2725107,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.816944,
           "population": "AFR",
           "refAlleleFreq": 0.183056,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.71153843,
           "population": "JPT",
           "refAlleleFreq": 0.28846157,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8657407,
           "population": "YRI",
           "refAlleleFreq": 0.13425928,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.080514,
           "population": "ALL",
           "refAlleleFreq": 0.919486,
           "study": "DISCOVEHR"
          },
          {
           "altAlleleFreq": 0.065656565,
           "population": "CEU",
           "refAlleleFreq": 0.93434346,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.07403958,
           "population": "EA",
           "refAlleleFreq": 0.9259604,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.25270948,
           "population": "ALL",
           "refAlleleFreq": 0.7472905,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.7700375,
           "population": "EAS",
           "refAlleleFreq": 0.22996253,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.765873,
           "population": "EAS",
           "refAlleleFreq": 0.23412699,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.21347032,
           "population": "OTH",
           "refAlleleFreq": 0.78652966,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.073525526,
           "population": "ALL",
           "refAlleleFreq": 0.92647445,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.26340505,
           "population": "FEMALE",
           "refAlleleFreq": 0.7365949,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.08124671,
           "population": "NFE",
           "refAlleleFreq": 0.91875327,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.71236926,
           "population": "AA",
           "refAlleleFreq": 0.28763074,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.07078903,
           "population": "TWINSUK_NODUP",
           "refAlleleFreq": 0.92921096,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.7655039,
           "population": "EAS",
           "refAlleleFreq": 0.23449612,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.075505964,
           "population": "ALSPAC",
           "refAlleleFreq": 0.924494,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.76229507,
           "population": "ASW",
           "refAlleleFreq": 0.23770493,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7705882,
           "population": "MSL",
           "refAlleleFreq": 0.22941178,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.3465511,
           "population": "AMR",
           "refAlleleFreq": 0.65344894,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.121212125,
           "population": "FIN",
           "refAlleleFreq": 0.8787879,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.75719637,
           "population": "EAS",
           "refAlleleFreq": 0.24280363,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.22596154,
           "population": "PUR",
           "refAlleleFreq": 0.77403843,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.35872236,
           "population": "AMR",
           "refAlleleFreq": 0.6412777,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.06513026,
           "population": "ALL",
           "refAlleleFreq": 0.93486977,
           "study": "GONL"
          },
          {
           "altAlleleFreq": 0.8362832,
           "population": "GWD",
           "refAlleleFreq": 0.1637168,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.104395606,
           "population": "GBR",
           "refAlleleFreq": 0.8956044,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.07009346,
           "population": "IBS",
           "refAlleleFreq": 0.92990655,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.35465014,
           "population": "AMR",
           "refAlleleFreq": 0.64534986,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.4883721,
           "population": "BEB",
           "refAlleleFreq": 0.5116279,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7552083,
           "population": "ACB",
           "refAlleleFreq": 0.24479169,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8636364,
           "population": "ESN",
           "refAlleleFreq": 0.13636363,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.07620284,
           "population": "NFE",
           "refAlleleFreq": 0.92379713,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.2287234,
           "population": "CLM",
           "refAlleleFreq": 0.7712766,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.16666667,
           "population": "OTH",
           "refAlleleFreq": 0.8333333,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.32355744,
           "population": "ALL",
           "refAlleleFreq": 0.67644256,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.30882353,
           "population": "ITU",
           "refAlleleFreq": 0.6911765,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.18516463,
           "population": "OTH",
           "refAlleleFreq": 0.81483537,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.08499929,
           "population": "NFE",
           "refAlleleFreq": 0.9150007,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.24660826,
           "population": "ALL",
           "refAlleleFreq": 0.75339174,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.7952381,
           "population": "CHS",
           "refAlleleFreq": 0.20476192,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.28645834,
           "population": "PJL",
           "refAlleleFreq": 0.7135416,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.09550562,
           "population": "ALL",
           "refAlleleFreq": 0.9044944,
           "study": "MGP"
          },
          {
           "altAlleleFreq": 0.34287626,
           "population": "SAS",
           "refAlleleFreq": 0.65712374,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.2882353,
           "population": "PEL",
           "refAlleleFreq": 0.7117647,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.69902915,
           "population": "CHB",
           "refAlleleFreq": 0.30097085,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.73203385,
           "population": "AFR",
           "refAlleleFreq": 0.26796615,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.4968051,
           "population": "ALL",
           "refAlleleFreq": 0.5031949,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.328125,
           "population": "MXL",
           "refAlleleFreq": 0.671875,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.15104775,
           "population": "FIN",
           "refAlleleFreq": 0.84895223,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.26080692,
           "population": "AMR",
           "refAlleleFreq": 0.7391931,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.41262135,
           "population": "GIH",
           "refAlleleFreq": 0.5873786,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.056074765,
           "population": "TSI",
           "refAlleleFreq": 0.94392526,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.29019094,
           "population": "ALL",
           "refAlleleFreq": 0.70980906,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.0714671,
           "population": "TWINSUK",
           "refAlleleFreq": 0.9285329,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.1628468,
           "population": "FIN",
           "refAlleleFreq": 0.8371532,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.41666666,
           "population": "STU",
           "refAlleleFreq": 0.5833334,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8548387,
           "population": "CDX",
           "refAlleleFreq": 0.14516127,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.08420836,
           "population": "ASJ",
           "refAlleleFreq": 0.91579163,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.23290326,
           "population": "MALE",
           "refAlleleFreq": 0.76709676,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.30839556,
           "population": "FEMALE",
           "refAlleleFreq": 0.69160444,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.7777778,
           "population": "KHV",
           "refAlleleFreq": 0.22222221,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.33579245,
           "population": "MALE",
           "refAlleleFreq": 0.6642076,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.1122449,
           "population": "ASJ",
           "refAlleleFreq": 0.8877551,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.82828283,
           "population": "LWK",
           "refAlleleFreq": 0.17171717,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.74387497,
           "population": "AFR",
           "refAlleleFreq": 0.25612503,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.14298809,
           "population": "FIN",
           "refAlleleFreq": 0.8570119,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.08250497,
           "population": "EUR",
           "refAlleleFreq": 0.917495,
           "study": "1kG_phase3"
          }
         ],
         "reference": "T",
         "repeat": [],
         "start": 1233941,
         "traitAssociation": [],
         "variantTraitAssociation": {},
         "xrefs": [
          {
           "id": "ENST00000379037",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000353662",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000354980",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000470659",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000438966",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000492936",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000492936(ENSG00000131584):n.2602+6A>G",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENSG00000131584",
           "source": "ensemblGene"
          },
          {
           "id": "ENST00000354700(ENSG00000131584):c.863+6A>G",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000354700",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000467278",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000479108",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000353662(ENSG00000131584):c.737+6A>G",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000476572",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000476572(ENSG00000131584):n.386+6A>G",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000493992",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000478065",
           "source": "ensemblTranscript"
          },
          {
           "id": "rs1739855",
           "source": "dbSNP"
          },
          {
           "id": "ENST00000467278(ENSG00000131584):n.389+6A>G",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000472541",
           "source": "ensemblTranscript"
          }
         ]
        },
        "chromosome": "1",
        "end": 1233941,
        "id": "1:1233941:T:C",
        "length": 1,
        "names": [],
        "reference": "T",
        "start": 1233941,
        "strand": "+",
        "studies": [],
        "type": "SNP"
       },
       {
        "alternate": "C",
        "annotation": {
         "alternate": "C",
         "chromosome": "1",
         "consequenceTypes": [
          {
           "biotype": "protein_coding",
           "cdnaPosition": 350,
           "cdsPosition": 305,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000520777",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 102,
            "reference": "MET",
            "substitutionScores": [
             {
              "description": "tolerated",
              "score": 0.12,
              "source": "sift"
             },
             {
              "description": "benign",
              "score": 0.001,
              "source": "polyphen"
             }
            ],
            "uniprotAccession": "Q96AX9",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001583",
             "name": "missense_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000355826",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000467597",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000378708",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "basic"
           ]
          },
          {
           "biotype": "processed_transcript",
           "cdnaPosition": 291,
           "cdsPosition": 0,
           "codon": "",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000512004",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001792",
             "name": "non_coding_transcript_exon_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000479659",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001619",
             "name": "non_coding_transcript_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ]
          },
          {
           "biotype": "nonsense_mediated_decay",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000502470",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001621",
             "name": "NMD_transcript_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000508455",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000477990",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001632",
             "name": "downstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000473511",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "cdnaPosition": 328,
           "cdsPosition": 305,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000518681",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 102,
            "reference": "MET",
            "substitutionScores": [
             {
              "description": "tolerated",
              "score": 0.09,
              "source": "sift"
             },
             {
              "description": "benign",
              "score": 0.001,
              "source": "polyphen"
             }
            ],
            "uniprotAccession": "Q96AX9",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001583",
             "name": "missense_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "protein_coding",
           "cdnaPosition": 350,
           "cdsPosition": 134,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000357210",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 45,
            "reference": "MET",
            "substitutionScores": [
             {
              "description": "tolerated",
              "score": 0.17,
              "source": "sift"
             },
             {
              "description": "benign",
              "score": 0,
              "source": "polyphen"
             }
            ],
            "uniprotAccession": "",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001583",
             "name": "missense_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "basic"
           ]
          },
          {
           "biotype": "nonsense_mediated_decay",
           "cdnaPosition": 353,
           "cdsPosition": 2,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000487053",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 1,
            "reference": "MET",
            "substitutionScores": [],
            "uniprotAccession": "",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0002012",
             "name": "start_lost"
            },
            {
             "accession": "SO:0001621",
             "name": "NMD_transcript_variant"
            }
           ]
          },
          {
           "biotype": "nonsense_mediated_decay",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000514363",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001621",
             "name": "NMD_transcript_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000378712",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "retained_intron",
           "cdnaPosition": 328,
           "cdsPosition": 0,
           "codon": "",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000507229",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001792",
             "name": "non_coding_transcript_exon_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000464570",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "cdnaPosition": 350,
           "cdsPosition": 134,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000378710",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 45,
            "reference": "MET",
            "substitutionScores": [
             {
              "description": "tolerated",
              "score": 0.18,
              "source": "sift"
             },
             {
              "description": "benign",
              "score": 0.004,
              "source": "polyphen"
             }
            ],
            "uniprotAccession": "",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001583",
             "name": "missense_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "basic"
           ]
          },
          {
           "biotype": "protein_coding",
           "cdnaPosition": 322,
           "cdsPosition": 305,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000505820",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 102,
            "reference": "MET",
            "substitutionScores": [
             {
              "description": "tolerated",
              "score": 0.12,
              "source": "sift"
             },
             {
              "description": "benign",
              "score": 0.007,
              "source": "polyphen"
             }
            ],
            "uniprotAccession": "Q96AX9",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001583",
             "name": "missense_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "CCDS",
            "basic"
           ]
          },
          {
           "biotype": "protein_coding",
           "cdnaPosition": 329,
           "cdsPosition": 2,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000504599",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 1,
            "reference": "MET",
            "substitutionScores": [
             {
              "description": "deleterious",
              "score": 0,
              "source": "sift"
             },
             {
              "description": "benign",
              "score": 0,
              "source": "polyphen"
             }
            ],
            "uniprotAccession": "",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0002012",
             "name": "start_lost"
            }
           ],
           "transcriptAnnotationFlags": [
            "basic"
           ]
          },
          {
           "biotype": "nonsense_mediated_decay",
           "cdnaPosition": 269,
           "cdsPosition": 134,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000489635",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 45,
            "reference": "MET",
            "substitutionScores": [],
            "uniprotAccession": "",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001583",
             "name": "missense_variant"
            },
            {
             "accession": "SO:0001621",
             "name": "NMD_transcript_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000503789",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "mRNA_end_NF",
            "cds_end_NF"
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000505370",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ]
          },
          {
           "biotype": "retained_intron",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000508148",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ]
          },
          {
           "biotype": "nonsense_mediated_decay",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000506488",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001621",
             "name": "NMD_transcript_variant"
            },
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000510793",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001627",
             "name": "intron_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "mRNA_end_NF",
            "cds_end_NF"
           ]
          },
          {
           "biotype": "protein_coding",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000514234",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001636",
             "name": "2KB_upstream_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "cds_start_NF",
            "mRNA_start_NF"
           ]
          },
          {
           "biotype": "nonsense_mediated_decay",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000486072",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001631",
             "name": "upstream_gene_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "cds_start_NF",
            "mRNA_start_NF"
           ]
          },
          {
           "biotype": "protein_coding",
           "cdnaPosition": 350,
           "cdsPosition": 134,
           "codon": "aTg/aCg",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000360522",
           "geneName": "MIB2",
           "proteinVariantAnnotation": {
            "alternate": "THR",
            "position": 45,
            "reference": "MET",
            "substitutionScores": [
             {
              "description": "tolerated",
              "score": 0.17,
              "source": "sift"
             },
             {
              "description": "benign",
              "score": 0.004,
              "source": "polyphen"
             }
            ],
            "uniprotAccession": "",
            "uniprotName": "",
            "uniprotVariantId": ""
           },
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001583",
             "name": "missense_variant"
            }
           ],
           "transcriptAnnotationFlags": [
            "basic"
           ]
          },
          {
           "biotype": "retained_intron",
           "cdnaPosition": 251,
           "cdsPosition": 0,
           "codon": "",
           "ensemblGeneId": "ENSG00000197530",
           "ensemblTranscriptId": "ENST00000511502",
           "geneName": "MIB2",
           "sequenceOntologyTerms": [
            {
             "accession": "SO:0001792",
             "name": "non_coding_transcript_exon_variant"
            }
           ]
          }
         ],
         "conservation": [
          {
           "description": "",
           "score": 0.1469999998807907,
           "source": "phylop"
          },
          {
           "description": "",
           "score": 0.37400001287460327,
           "source": "phastCons"
          },
          {
           "description": "",
           "score": -2.4700000286102295,
           "source": "gerp"
          }
         ],
         "cytoband": [
          {
           "chromosome": "1",
           "end": 2300000,
           "name": "p36.33",
           "stain": "gneg",
           "start": 1
          }
         ],
         "displayConsequenceType": "start_lost",
         "end": 1558792,
         "functionalScore": [
          {
           "description": "",
           "score": -1.0699996948242188,
           "source": "cadd_raw"
          },
          {
           "description": "",
           "score": 0.009999999776482582,
           "source": "cadd_scaled"
          }
         ],
         "hgvs": [
          "ENST00000520777(ENSG00000197530):c.305T>C",
          "ENST00000357210(ENSG00000197530):c.134T>C",
          "ENST00000360522(ENSG00000197530):c.134T>C",
          "ENST00000378710(ENSG00000197530):c.134T>C",
          "ENST00000355826(ENSG00000197530):c.282-19T>C",
          "ENST00000518681(ENSG00000197530):c.305T>C",
          "ENST00000505820(ENSG00000197530):c.305T>C",
          "ENST00000479659(ENSG00000197530):n.288-19T>C",
          "ENST00000489635(ENSG00000197530):c.134T>C",
          "ENST00000511502(ENSG00000197530):n.251T>C",
          "ENST00000514363(ENSG00000197530):c.111-19T>C",
          "ENST00000487053(ENSG00000197530):c.2T>C",
          "ENST00000502470(ENSG00000197530):c.-22-19T>C",
          "ENST00000378712(ENSG00000197530):c.-22-19T>C",
          "ENST00000510793(ENSG00000197530):c.-22-19T>C",
          "ENST00000504599(ENSG00000197530):c.2T>C",
          "ENST00000507229(ENSG00000197530):n.328T>C",
          "ENST00000503789(ENSG00000197530):c.-22-19T>C",
          "ENST00000378708(ENSG00000197530):c.-22-19T>C",
          "ENST00000506488(ENSG00000197530):c.-22-19T>C",
          "ENST00000512004(ENSG00000197530):n.291T>C"
         ],
         "populationFrequencies": [
          {
           "altAlleleFreq": 0.8599182,
           "population": "SAS",
           "refAlleleFreq": 0.14008182,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.750748,
           "population": "AFR",
           "refAlleleFreq": 0.24925202,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.70574886,
           "population": "AFR",
           "refAlleleFreq": 0.29425114,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.5480769,
           "population": "JPT",
           "refAlleleFreq": 0.45192307,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.6527778,
           "population": "YRI",
           "refAlleleFreq": 0.3472222,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.935193,
           "population": "ALL",
           "refAlleleFreq": 0.064807,
           "study": "DISCOVEHR"
          },
          {
           "altAlleleFreq": 0.9444444,
           "population": "CEU",
           "refAlleleFreq": 0.055555582,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9400635,
           "population": "EA",
           "refAlleleFreq": 0.059936523,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.8796547,
           "population": "ALL",
           "refAlleleFreq": 0.120345294,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.57080764,
           "population": "EAS",
           "refAlleleFreq": 0.42919236,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.5515873,
           "population": "EAS",
           "refAlleleFreq": 0.44841272,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.91151685,
           "population": "OTH",
           "refAlleleFreq": 0.088483155,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.93136734,
           "population": "ALL",
           "refAlleleFreq": 0.06863266,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.8579323,
           "population": "FEMALE",
           "refAlleleFreq": 0.14206767,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.94106144,
           "population": "NFE",
           "refAlleleFreq": 0.058938563,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.76677316,
           "population": "AA",
           "refAlleleFreq": 0.23322684,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.9266928,
           "population": "TWINSUK_NODUP",
           "refAlleleFreq": 0.073307216,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.5645963,
           "population": "EAS",
           "refAlleleFreq": 0.4354037,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.9346134,
           "population": "ALSPAC",
           "refAlleleFreq": 0.06538659,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.7295082,
           "population": "ASW",
           "refAlleleFreq": 0.27049178,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.69411767,
           "population": "MSL",
           "refAlleleFreq": 0.30588233,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.80965275,
           "population": "AMR",
           "refAlleleFreq": 0.19034725,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.8787879,
           "population": "FIN",
           "refAlleleFreq": 0.121212125,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.6240929,
           "population": "EAS",
           "refAlleleFreq": 0.37590712,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.8942308,
           "population": "PUR",
           "refAlleleFreq": 0.10576922,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.80861247,
           "population": "AMR",
           "refAlleleFreq": 0.19138753,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.93286574,
           "population": "ALL",
           "refAlleleFreq": 0.06713426,
           "study": "GONL"
          },
          {
           "altAlleleFreq": 0.70353985,
           "population": "GWD",
           "refAlleleFreq": 0.29646015,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9230769,
           "population": "GBR",
           "refAlleleFreq": 0.07692307,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9579439,
           "population": "IBS",
           "refAlleleFreq": 0.042056084,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8125373,
           "population": "AMR",
           "refAlleleFreq": 0.18746269,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.84302324,
           "population": "BEB",
           "refAlleleFreq": 0.15697676,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.703125,
           "population": "ACB",
           "refAlleleFreq": 0.296875,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7070707,
           "population": "ESN",
           "refAlleleFreq": 0.2929293,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9381251,
           "population": "NFE",
           "refAlleleFreq": 0.061874926,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.89361703,
           "population": "CLM",
           "refAlleleFreq": 0.106382966,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8965164,
           "population": "OTH",
           "refAlleleFreq": 0.10348362,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.8552759,
           "population": "ALL",
           "refAlleleFreq": 0.14472407,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.8480392,
           "population": "ITU",
           "refAlleleFreq": 0.15196079,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.90386194,
           "population": "OTH",
           "refAlleleFreq": 0.09613806,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.93561774,
           "population": "NFE",
           "refAlleleFreq": 0.064382255,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.86788756,
           "population": "ALL",
           "refAlleleFreq": 0.13211244,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.51428574,
           "population": "CHS",
           "refAlleleFreq": 0.48571426,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9166667,
           "population": "PJL",
           "refAlleleFreq": 0.08333331,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8604965,
           "population": "SAS",
           "refAlleleFreq": 0.13950348,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.87058824,
           "population": "PEL",
           "refAlleleFreq": 0.12941176,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.5776699,
           "population": "CHB",
           "refAlleleFreq": 0.42233008,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.7672843,
           "population": "AFR",
           "refAlleleFreq": 0.23271573,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.7735623,
           "population": "ALL",
           "refAlleleFreq": 0.22643769,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.78125,
           "population": "MXL",
           "refAlleleFreq": 0.21875,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8976418,
           "population": "FIN",
           "refAlleleFreq": 0.10235822,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.86743516,
           "population": "AMR",
           "refAlleleFreq": 0.13256484,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.86893207,
           "population": "GIH",
           "refAlleleFreq": 0.13106793,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.97196263,
           "population": "TSI",
           "refAlleleFreq": 0.02803737,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8855876,
           "population": "ALL",
           "refAlleleFreq": 0.11441243,
           "study": "ESP6500"
          },
          {
           "altAlleleFreq": 0.92799354,
           "population": "TWINSUK",
           "refAlleleFreq": 0.072006464,
           "study": "UK10K"
          },
          {
           "altAlleleFreq": 0.8940264,
           "population": "FIN",
           "refAlleleFreq": 0.1059736,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.8235294,
           "population": "STU",
           "refAlleleFreq": 0.17647058,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.5860215,
           "population": "CDX",
           "refAlleleFreq": 0.41397852,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.9772678,
           "population": "ASJ",
           "refAlleleFreq": 0.022732198,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.87581134,
           "population": "MALE",
           "refAlleleFreq": 0.12418866,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.8623262,
           "population": "FEMALE",
           "refAlleleFreq": 0.1376738,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.53535354,
           "population": "KHV",
           "refAlleleFreq": 0.46464646,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.8495596,
           "population": "MALE",
           "refAlleleFreq": 0.1504404,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.98675495,
           "population": "ASJ",
           "refAlleleFreq": 0.013245046,
           "study": "GNOMAD_GENOMES"
          },
          {
           "altAlleleFreq": 0.7626263,
           "population": "LWK",
           "refAlleleFreq": 0.23737371,
           "study": "1kG_phase3"
          },
          {
           "altAlleleFreq": 0.74422014,
           "population": "AFR",
           "refAlleleFreq": 0.25577986,
           "study": "GNOMAD_EXOMES"
          },
          {
           "altAlleleFreq": 0.9003822,
           "population": "FIN",
           "refAlleleFreq": 0.09961778,
           "study": "EXAC"
          },
          {
           "altAlleleFreq": 0.9363817,
           "population": "EUR",
           "refAlleleFreq": 0.0636183,
           "study": "1kG_phase3"
          }
         ],
         "reference": "T",
         "repeat": [
          {
           "chromosome": "1",
           "copyNumber": 2,
           "end": 1558840,
           "id": "null",
           "percentageMatch": 0.95,
           "source": "trf",
           "start": 1558755
          }
         ],
         "start": 1558792,
         "traitAssociation": [
          {
           "bibliography": [],
           "ethnicity": "Z",
           "heritableTraits": [],
           "id": "COSM7229018",
           "somaticInformation": {
            "histologySubtype": "null",
            "primaryHistology": "null"
           },
           "source": {
            "name": "cosmic"
           },
           "submissions": []
          },
          {
           "bibliography": [],
           "ethnicity": "Z",
           "heritableTraits": [],
           "id": "COSM7229017",
           "somaticInformation": {
            "histologySubtype": "null",
            "primaryHistology": "null"
           },
           "source": {
            "name": "cosmic"
           },
           "submissions": []
          }
         ],
         "variantTraitAssociation": {
          "cosmic": [
           {
            "histologySubtype": "null",
            "mutationId": "COSM7229018",
            "primaryHistology": "null"
           },
           {
            "histologySubtype": "null",
            "mutationId": "COSM7229017",
            "primaryHistology": "null"
           }
          ]
         },
         "xrefs": [
          {
           "id": "ENST00000378708(ENSG00000197530):c.-22-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENSG00000197530",
           "source": "ensemblGene"
          },
          {
           "id": "ENST00000360522(ENSG00000197530):c.134T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000467597",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000512004",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000479659",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000477990",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000510793(ENSG00000197530):c.-22-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000473511",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000518681",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000357210",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000511502(ENSG00000197530):n.251T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000487053",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000514363",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000507229",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000378710(ENSG00000197530):c.134T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000514363(ENSG00000197530):c.111-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000504599",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000378712(ENSG00000197530):c.-22-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000489635",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000503789",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000505370",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000520777(ENSG00000197530):c.305T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000510793",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000514234",
           "source": "ensemblTranscript"
          },
          {
           "id": "rs12755088",
           "source": "dbSNP"
          },
          {
           "id": "ENST00000489635(ENSG00000197530):c.134T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000505820(ENSG00000197530):c.305T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000504599(ENSG00000197530):c.2T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000487053(ENSG00000197530):c.2T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000503789(ENSG00000197530):c.-22-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000506488(ENSG00000197530):c.-22-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000479659(ENSG00000197530):n.288-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000520777",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000355826",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000507229(ENSG00000197530):n.328T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000378708",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000518681(ENSG00000197530):c.305T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000502470",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000357210(ENSG00000197530):c.134T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000508455",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000512004(ENSG00000197530):n.291T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000355826(ENSG00000197530):c.282-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000378712",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000502470(ENSG00000197530):c.-22-19T>C",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000464570",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000378710",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000505820",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000508148",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000506488",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000486072",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000360522",
           "source": "ensemblTranscript"
          },
          {
           "id": "ENST00000511502",
           "source": "ensemblTranscript"
          }
         ]
        },
        "chromosome": "1",
        "end": 1558792,
        "id": "1:1558792:T:C",
        "length": 1,
        "names": [],
        "reference": "T",
        "start": 1558792,
        "strand": "+",
        "studies": [],
        "type": "SNV"
       }
      ],
      "text/plain": [
       "<IPython.core.display.JSON object>"
      ]
     },
     "execution_count": 24,
     "metadata": {
      "application/json": {
       "expanded": false,
       "root": "root"
      }
     },
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Filter by misense variants\n",
    "variants = oc.variants.query(study=study, ct='missense_variant', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by missense_variant:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by misense variants and stop_gained\n",
    "variants = oc.variants.query(study=study, ct='missense_variant,stop_gained', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by missense_variant,stop_gained:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by misense variants and stop_gained\n",
    "variants = oc.variants.query(study=study, ct='lof', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by lof:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Filter by Population Frequencies\n",
    "\n",
    "OpenCGA allows to filter variants by population frequencies, including:\n",
    "- Minor Allele frequency (MAF) with the parameter `populationFrequencyMaf`\n",
    "- Alternate Allele frequency (ALT) with the parameter `populationFrequencyAlt`\n",
    "<!-- - Reference Allele frequency with the parameter `populationFrequencyRef` -->\n",
    "\n",
    "\n",
    "The population frequency studies indexed in OpenCGA include different sources such as **genomAD** or **1000 Genomes**.\n",
    "\n",
    "The syntax for the query paramter is: `{study}:{population}:{cohort}[<|>|<=|>=]{proportion}`. Note that you can specify several populations separated by comma (OR) or by semi-colon (AND), e.g. for all varaints less than 5% in two studies we shuold use `1kG_phase3:ALL<0.01;GNOMAD_GENOMES:ALL<0.01`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by population freq MAF:\n",
      "-----------------------------------\n",
      "#Time: 283\n",
      "#Num matches: 47138\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:546952:T:C\t,RP5-857K21.4\tintron_variant\n",
      "1:745347:T:C\t,RP11-206L10.9,RP11-206L10.10,RP11-206L10.8\tnon_coding_transcript_exon_variant\n",
      "1:745371:A:-\t,RP11-206L10.9,RP11-206L10.10,RP11-206L10.8\tnon_coding_transcript_exon_variant\n",
      "1:14907:A:G\t,WASH7P,DDX11L1\tintron_variant\n",
      "1:14948:G:A\t,WASH7P,DDX11L1\tintron_variant\n",
      "\n",
      "\n",
      "Filter by population freq MAF:\n",
      "-----------------------------------\n",
      "#Time: 283\n",
      "#Num matches: 33057\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:63336:C:T\t,OR4G11P\tnon_coding_transcript_exon_variant\n",
      "1:936848:G:C\tHES4,,RP11-54O7.17\t2KB_upstream_variant\n",
      "1:1001178:T:G\t,RP11-54O7.18,RP11-465B22.3\tintron_variant\n",
      "1:1162673:G:C\tB3GALT6,,SDF4\tnon_coding_transcript_exon_variant\n",
      "1:1194356:T:C\t,UBE2J2\tintron_variant\n",
      "\n",
      "\n",
      "Filter by population freq ALT:\n",
      "-----------------------------------\n",
      "#Time: 275\n",
      "#Num matches: 6835\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:63336:C:T\t,OR4G11P\tnon_coding_transcript_exon_variant\n",
      "1:936848:G:C\tHES4,,RP11-54O7.17\t2KB_upstream_variant\n",
      "1:1179800:G:A\tFAM132A,\tsynonymous_variant\n",
      "1:1226690:C:T\t,ACAP3,SCNN1D\tsynonymous_variant\n",
      "1:1294411:C:T\t,MXRA8\tintron_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter by population frequency maf\n",
    "population_frequency_alt='1kG_phase3:ALL<0.01'\n",
    "variants = oc.variants.query(study=study, populationFrequencyAlt=population_frequency_alt, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by population freq MAF:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by two population frequency maf\n",
    "## Remember to use commas for OR and semi-colon for AND\n",
    "population_frequency_alt='1kG_phase3:ALL<0.01;GNOMAD_GENOMES:ALL<0.01'\n",
    "variants = oc.variants.query(study=study, populationFrequencyAlt=population_frequency_alt, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by population freq MAF:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by population frequency alternate\n",
    "population_frequency_alt='1kG_phase3:ALL>0;1kG_phase3:ALL<0.01'\n",
    "variants = oc.variants.query(study=study, populationFrequencyAlt=population_frequency_alt, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by population freq ALT:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Filter by Cohort Stats\n",
    "\n",
    "OpenCGA allows users to define cohorts of samples and calculate and index the allele and genotype frequencies among other stats. By default, a cohosrt called **ALL** containing all samples is defined and the variant stats are calculated.\n",
    "You can filter by the internal cohort stats using the parameter `cohortStatsAlt`, e.g. `{study}:ALL<0.0001`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by Cohort Stats:\n",
      "----------------------------\n",
      "#Time: 272\n",
      "#Num matches: 0\n",
      "#Num results: 0\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter by Cohort Stats\n",
    "cohort_stats = study + ':ALL<0.0001'\n",
    "variants = oc.variants.query(study=study, cohortStatsAlt=cohort_stats, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Cohort Stats:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. Filter by Clinical Information\n",
    "\n",
    "OpenCGA allows to filter variants by ClinVar clinical significance using the parameter `clinicalSignificance`.\n",
    "The accepted values are: **benign, likely_benign, uncertain_significance, likely_pathogenic, pathogenic**.\n",
    "\n",
    "There are also other parameters like `geneTraitId`, which accept list of gene trait association ids, e.g: **\"umls:C0007222\" , \"OMIM:269600\"**."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by Clinical Significances:\n",
      "--------------------------------------\n",
      "#Time: 279\n",
      "#Num matches: 55\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "2:227892720:C:T\tCOL4A4\tmissense_variant\n",
      "22:18905964:C:T\tDGCR6,,PRODH\tmissense_variant\n",
      "3:39307162:G:A\t,CX3CR1\tmissense_variant\n",
      "3:39307256:C:T\t,CX3CR1\tmissense_variant\n",
      "4:187158034:G:A\tKLKB1\tmissense_variant\n",
      "\n",
      "\n",
      "Filter by population gene traits\n",
      "-------------------------------------\n",
      "#Time: 303\n",
      "#Num matches: 154\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:45795084:G:A\t,HPDL,MUTYH\tmissense_variant\n",
      "1:45796269:G:C\t,HPDL,MUTYH\tintron_variant\n",
      "1:45797505:C:G\t,HPDL,MUTYH\tmissense_variant\n",
      "1:45798555:T:C\t,HPDL,MUTYH\tnon_coding_transcript_exon_variant\n",
      "1:45805566:G:C\tTOE1,,MUTYH,TESK2\tsplice_region_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter by Clinical Significance\n",
    "clinical_significance = 'likely_pathogenic,pathogenic'\n",
    "variants = oc.variants.query(study=study, clinicalSignificance=clinical_significance, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Clinical Significances:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by population gene trait \n",
    "gene_trait_id = 'OMIM:137215,OMIM:613659'  #GASTRIC CANCER terms\n",
    "variants = oc.variants.query(study=study, geneTraitId=gene_trait_id, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by population gene traits') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 8. Disease Panel\n",
    "\n",
    "OpenCGA allows users to define and store disease panels with genes and variants, you can easily filter by **disease panel** using the parameter `panel`, e.g. `panel=Amyloidosis`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by Disease Panel:\n",
      "-----------------------------\n",
      "#Time: 303\n",
      "#Num matches: 78\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "20:23607874:T:G\tCST3,RP11-218C14.8,\t2KB_downstream_variant\n",
      "20:23610375:G:C\tCST3,RP11-218C14.8,\tintron_variant\n",
      "20:23615820:-:CA\tCST3,RP11-218C14.8,\tintron_variant\n",
      "20:23616996:A:T\tCST3,RP11-218C14.8,\tintron_variant\n",
      "20:23618395:T:C\tCST3,\tsynonymous_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter by Clinical Significance\n",
    "variants = oc.variants.query(study=study, panel='Amyloidosis-PanelAppId-502', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Disease Panel:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 9. Bulding Complex Queries\n",
    "\n",
    "OpenCGA implements a very advanced variant query engine that allows to combine many filters to build very complex and useful queries. In this section you will find some examples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by Consequence Type and Disease Panel:\n",
      "--------------------------------------------------\n",
      "#Time: 281\n",
      "#Num matches: 1\n",
      "#Num results: 1\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "12:69744014:C:A\tLYZ,,RP11-1143G9.4\tmissense_variant\n",
      "\n",
      "\n",
      "Filter by Consequence Type and Clinical Significance:\n",
      "----------------------------------------------------------\n",
      "#Time: 275\n",
      "#Num matches: 34\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "22:18905964:C:T\tDGCR6,,PRODH\tmissense_variant\n",
      "3:39307162:G:A\t,CX3CR1\tmissense_variant\n",
      "3:39307256:C:T\t,CX3CR1\tmissense_variant\n",
      "4:187158034:G:A\tKLKB1\tmissense_variant\n",
      "3:38645420:T:C\t,SCN5A\tmissense_variant\n",
      "\n",
      "\n",
      "Filter by Consequence Type, Clinical Significance and Cohort Satats:\n",
      "-------------------------------------------------------------------------\n",
      "#Time: 276\n",
      "#Num matches: 0\n",
      "#Num results: 0\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "\n",
      "\n",
      "Filter by Consequence Type, Clinical Significance and Cohort Satats:\n",
      "-------------------------------------------------------------------------\n",
      "#Time: 254\n",
      "#Num matches: 0\n",
      "#Num results: 0\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter by Consequence Type and Disease Panel\n",
    "variants = oc.variants.query(study=study, ct='lof,missense_variant', panel='Amyloidosis-PanelAppId-502', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Consequence Type and Disease Panel:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by Consequence Type and Clinical Significance\n",
    "variants = oc.variants.query(study=study, ct='lof,missense_variant', clinicalSignificance='pathogenic', exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Consequence Type and Clinical Significance:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by Consequence Type, Clinical Significance and Cohort Satats\n",
    "cohort_stats = study + ':ALL<0.005'\n",
    "variants = oc.variants.query(study=study, ct='lof,missense_variant', clinicalSignificance='likely_pathogenic,pathogenic', cohortStatsAlt=cohort_stats, exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Consequence Type, Clinical Significance and Cohort Satats:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by Consequence Type, Clinical Significance and Cohort Satats\n",
    "populationFrequencyAlt='1kG_phase3:ALL<0.01;GNOMAD_GENOMES:ALL<0.01'\n",
    "variants = oc.variants.query(study=study, ct='lof,missense_variant', clinicalSignificance='likely_pathogenic,pathogenic', cohortStatsAlt=cohort_stats, populationFrequencyMaf=populationFrequencyAlt\n",
    ", exclude='studies', limit=5) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by Consequence Type, Clinical Significance and Cohort Satats:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Querying Variants: Filter by Sample Data\n",
    "----------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Filter By Sample\n",
    "\n",
    "OpenCGA implements different parameters to allow users to filter variants for a given sample or set of samples, you just need to use the parameter `sample`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample IDs for the next examples are: ISDBM322016 and ISDBM322017\n",
      "\n"
     ]
    }
   ],
   "source": [
    "resp = oc.samples.search(study=study, limit=2)\n",
    "sample_id1 = resp.get_result(0)['id']\n",
    "sample_id2 = resp.get_result(1)['id']\n",
    "# JSON(resp.get_results())\n",
    "print('Sample IDs for the next examples are: {} and {}\\n'.format(sample_id1, sample_id2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by sample ISDBM322016:\n",
      "----------------------------------\n",
      "#Time: 406\n",
      "#Num matches: 164776\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:14907:A:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:14930:A:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:15211:T:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:17538:C:A\tWASH7P,DDX11L1\tintron_variant\n",
      "1:63336:C:T\tOR4G11P\tnon_coding_transcript_exon_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Fetch all variants for one sample\n",
    "variants = oc.variants.query(study=study, sample=sample_id1, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by sample {}:'.format(sample_id1)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Filter by Genotype\n",
    "\n",
    "One of the most useful functionality is filter by genotype, you just need to add the genoype to the `sample` parameter.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by sample ISDBM322016:0/1:\n",
      "--------------------------------------\n",
      "#Time: 359\n",
      "#Num matches: 68280\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:14907:A:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:14930:A:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:15211:T:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:17538:C:A\tWASH7P,DDX11L1\tintron_variant\n",
      "1:63336:C:T\tOR4G11P\tnon_coding_transcript_exon_variant\n",
      "\n",
      "\n",
      "Filter by sample ISDBM322016:1/1:\n",
      "--------------------------------------\n",
      "#Time: 298\n",
      "#Num matches: 96044\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:63736:CTA:-\tOR4G11P\tnon_coding_transcript_exon_variant\n",
      "1:69511:A:G\tOR4F5\tmissense_variant\n",
      "1:752566:G:A\tFAM87B,RP11-206L10.10\tintron_variant\n",
      "1:752721:A:G\tFAM87B,RP11-206L10.10\tintron_variant\n",
      "1:753405:C:A\tFAM87B,RP11-206L10.10\tnon_coding_transcript_exon_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter HET variants\n",
    "genotype = sample_id1 + ':0/1'\n",
    "variants = oc.variants.query(study=study, sample=genotype, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by sample {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter HOM variants\n",
    "genotype = sample_id1 + ':1/1'\n",
    "variants = oc.variants.query(study=study, sample=genotype, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by sample {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You can also filter using **more than sample** and query using OR or AND."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by samples ISDBM322016:1/1,ISDBM322017:0/1,1/1:\n",
      "-----------------------------------------------------------\n",
      "#Time: 734\n",
      "#Num matches: 184240\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:14907:A:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:14930:A:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:14948:G:A\tWASH7P,DDX11L1\tintron_variant\n",
      "1:15211:T:G\tWASH7P,DDX11L1\tintron_variant\n",
      "1:63336:C:T\tOR4G11P\tnon_coding_transcript_exon_variant\n",
      "\n",
      "\n",
      "Filter by samples ISDBM322016:1/1;ISDBM322017:0/1,1/1:\n",
      "-----------------------------------------------------------\n",
      "#Time: 535\n",
      "#Num matches: 55734\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:63736:CTA:-\tOR4G11P\tnon_coding_transcript_exon_variant\n",
      "1:69511:A:G\tOR4F5\tmissense_variant\n",
      "1:752566:G:A\tFAM87B,RP11-206L10.10\tintron_variant\n",
      "1:752721:A:G\tFAM87B,RP11-206L10.10\tintron_variant\n",
      "1:753405:C:A\tFAM87B,RP11-206L10.10\tnon_coding_transcript_exon_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter all HOM variants for two samples\n",
    "genotype = sample_id1 + ':1/1' + ',' + sample_id2 + ':0/1,1/1'\n",
    "variants = oc.variants.query(study=study, sample=genotype, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by samples {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter common HOM variants for two samples\n",
    "genotype = sample_id1 + ':1/1' + ';' + sample_id2 + ':0/1,1/1'\n",
    "variants = oc.variants.query(study=study, sample=genotype, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by samples {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Filter by Mode of Inheritance\n",
    "\n",
    "OpenCGA also implemets queris by different seggregation modes such as recessive, compound heterozygous or de Novo.\n",
    "\n",
    "**IMPORTANT NOTE** This only works for families!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter by autosomalRecessive\n",
    "genotype = sample_id1 + ':autosomalRecessive'\n",
    "variants = oc.variants.query(study=study, sample=genotype, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by autosomalRecessive {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by compoundHeterozygous\n",
    "genotype = sample_id1 + ':compoundHeterozygous'\n",
    "variants = oc.variants.query(study=study, sample=genotype, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by compoundHeterozygous {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by deNovo\n",
    "genotype = sample_id1 + ':deNovo'\n",
    "variants = oc.variants.query(study=study, sample=genotype, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by deNovo {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Filter by VCF data\n",
    "\n",
    "You can also filter by any field in the VCF file, one of the most interesting is filtering by PASS variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All 1/1 variants for sample ISDBM322016:1/1:\n",
      "-------------------------------------------------\n",
      "#Time: 324\n",
      "#Num matches: 96044\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:63736:CTA:-\tOR4G11P\tnon_coding_transcript_exon_variant\n",
      "1:69511:A:G\tOR4F5\tmissense_variant\n",
      "1:752566:G:A\tFAM87B,RP11-206L10.10\tintron_variant\n",
      "1:752721:A:G\tFAM87B,RP11-206L10.10\tintron_variant\n",
      "1:753405:C:A\tFAM87B,RP11-206L10.10\tnon_coding_transcript_exon_variant\n",
      "\n",
      "\n",
      "PASS 1/1 variants for sample ISDBM322016:1/1:\n",
      "--------------------------------------------------\n",
      "#Time: 489\n",
      "#Num matches: 89302\n",
      "#Num results: 5\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "1:63736:CTA:-\tOR4G11P\tnon_coding_transcript_exon_variant\n",
      "1:752566:G:A\tFAM87B,RP11-206L10.10\tintron_variant\n",
      "1:752721:A:G\tFAM87B,RP11-206L10.10\tintron_variant\n",
      "1:808631:G:A\tTUBB8P11,FAM41C\tintron_variant\n",
      "1:812267:A:G\tTUBB8P11,FAM41C\tnon_coding_transcript_exon_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter HOM variants\n",
    "genotype = sample_id1 + ':1/1'\n",
    "variants = oc.variants.query(study=study, sample=genotype, limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='All 1/1 variants for sample {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter PASS HOM variants\n",
    "variants = oc.variants.query(study=study, sample=genotype, filter='PASS', limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='PASS 1/1 variants for sample {}:'.format(genotype)) # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Building Complex queries\n",
    "\n",
    "OpenGCA allows you to combine different filters to build more complex queries."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by genotype, genes and clinical significance\n",
      "--------------------------------------------------------\n",
      "#Time: 431\n",
      "#Num matches: 0\n",
      "#Num results: 0\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "\n",
      "\n",
      "Filter by genotype, genes, population frequencies and clinical significance:\n",
      "---------------------------------------------------------------------------------\n",
      "#Time: 996\n",
      "#Num matches: 0\n",
      "#Num results: 0\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Filter by genotype, genes and clinical significance\n",
    "genotype = sample_id1 + ':1/1'\n",
    "variants = oc.variants.query(study=study, sample=genotype, gene='BRCA2,TP53', clinicalSignificance='likely_pathogenic,pathogenic', limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by genotype, genes and clinical significance') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Filter by genotype, genes, population frequencies and clinical significance\n",
    "population_frequency_alt='1kG_phase3:ALL<0.01;GNOMAD_GENOMES:ALL<0.01'\n",
    "variants = oc.variants.query(study=study, sample=genotype, gene='BRCA2,TP53', populationFrequencyAlt=population_frequency_alt, clinicalSignificance='likely_pathogenic,pathogenic', limit=5, count=True) # Other params: exclude='annotation'\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by genotype, genes, population frequencies and clinical significance:') # metadata=False\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Aggregation Stats\n",
    "-------\n",
    "\n",
    "OpenCGA implements an advanced aggregation functionality that allows to get stats of the variants."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Variant Aggregation Stats using Variant Annotation\n",
    "\n",
    "You can easily filter variants (_optional_) and select different fields to aggregate and count variants. You can get several aggregations in the same executin. This also allows to nest fields using the character `>>`.\n",
    "\n",
    "**Important Note**: to make the notebook more responsive we are filtering variants in the following examples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Aggregate by type all the BRCA2 variants\n",
    "variants = oc.variants.aggregation_stats(study=study, gene='BRCA2', fields=\"type\", limit=5) # Other params: exclude='annotation'\n",
    "# pprint(variants.get_responses())\n",
    "# print(RESULT_SEPARATOR)\n",
    "\n",
    "## Aggregate by chromsome and type all the lof and missense variants\n",
    "variants = oc.variants.aggregation_stats(study=study, ct='lof,missense_variant', fields=\"chromosome;type\", limit=5) # Other params: exclude='annotation'\n",
    "# pprint(variants.get_responses())\n",
    "# print(RESULT_SEPARATOR)\n",
    "\n",
    "## Aggregate by type in each chromsome all the lof and missense variants\n",
    "variants = oc.variants.aggregation_stats(study=study, ct='lof,missense_variant', fields=\"chromosome>>type\", limit=5) # Other params: exclude='annotation'\n",
    "# pprint(variants.get_responses())\n",
    "# print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Variant Aggregation Stats by Sample\n",
    "\n",
    "OpenCGA index all sample data in a very efficient way, this allows OpenCGA to execute aggregations for a specific sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Aggregate by type all the BRCA2 variants\n",
    "variants = oc.variants.aggregation_stats_sample(study=study, sample=sample_id1, fields='type', limit=5) # Other params: exclude='annotation'\n",
    "# pprint(variants.get_responses())\n",
    "# print(RESULT_SEPARATOR)\n",
    "\n",
    "## Aggregate by chromsome and type all the lof and missense variants\n",
    "variants = oc.variants.aggregation_stats_sample(study=study, sample=sample_id1, ct='lof,missense_variant', fields=\"chromosome;type\", limit=5) # Other params: exclude='annotation'\n",
    "# pprint(variants.get_responses())\n",
    "# print(RESULT_SEPARATOR)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variants.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Use Cases\n",
    "---------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Fetch all Samples for a given Variant\n",
    "\n",
    "A very common use case is to fetch all samples togteher with the clinical data for a given variant of interest.\n",
    "\n",
    "In this section we are going to choose randomely one variant and then fecth samples and clinical data, these are the steps:\n",
    "\n",
    "1. Choose one random variant\n",
    "2. Fetch the all the samples with a genotye 1/1\n",
    "3. Query _Catalog_ to retrieve the clinical data of the samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filter by genotype, genes and clinical significance\n",
      "--------------------------------------------------------\n",
      "#Time: 407\n",
      "#Num matches: 40\n",
      "#Num results: 1\n",
      "#Num inserted: 0\n",
      "#Num updated: 0\n",
      "#Num deleted: 0\n",
      "#id\tannotation.consequenceTypes.geneName\tannotation.displayConsequenceType\n",
      "22:18905964:C:T\tDGCR6,PRODH\tmissense_variant\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Step 1 - Select one random variant for this example\n",
    "variants = oc.variants.query(study=study, clinicalSignificance='pathogenic', limit=1, count=True) # Other params: exclude='annotation'\n",
    "\n",
    "## Store selected variant ID\n",
    "variant_id = variants.get_result(0)['id']\n",
    "\n",
    "## Print basic information\n",
    "variants.print_results(fields='id,annotation.consequenceTypes.geneName,annotation.displayConsequenceType', title='Filter by genotype, genes and clinical significance') # metadata=False\n",
    "print(RESULT_SEPARATOR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [],
   "source": [
    "## Step 2 - Fetch the first 10 samples with the genoype 0/1 or 1/1\n",
    "genotype='0/1,1/1'\n",
    "limit=10\n",
    "variants = oc.variants.query_sample(study=study, variant=variant_id, genotype=genotype, limit=limit) # Other params: exclude='annotation'\n",
    "\n",
    "# Uncomment to dislay samples in JSON viewer\n",
    "# JSON(variants.get_result(0)['studies'][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ISDBM322015', 'ISDBM322018']\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Step 3 - Fetch Clinical data\n",
    "samples = variants.get_result(0)['studies'][0]['samples']\n",
    "\n",
    "# Get all sampled IDs\n",
    "sample_ids = []\n",
    "for sample in samples:\n",
    "    sample_ids.append(sample['sampleId'])\n",
    "print(sample_ids)\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "# Featch data from Catalog\n",
    "clinical_data = oc.samples.search(study=study, id=','.join(sample_ids), includeIndividual=True)\n",
    "# JSON(clinical_data.get_results())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "source": [
    "## 2. Sample Variant Stats\n",
    "\n",
    "You can alse get rich variants stats for a given sample. In this section we will:\n",
    "\n",
    "1. Choose one random sample from _Catalog_\n",
    "2. Calculate sample variant stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ISDBM322016\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Step 1 - Fetch a rondom sample\n",
    "resp = oc.samples.search(study=study, limit=2)\n",
    "sample_id = resp.get_result(0)['id']\n",
    "print(sample_id)\n",
    "print(RESULT_SEPARATOR)\n",
    "\n",
    "## Step 2 - Calculate Variant Stats\n",
    "variant_stats = oc.variants.query_sample_stats(study=study, sample=sample_id)\n",
    "\n",
    "## Uncomment next line to display an interactive JSON viewer\n",
    "# JSON(variant_stats.get_results())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
